*** ./smali/android/graphics/drawable/NinePatchDrawable.smali	2013-03-18 17:23:43.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/graphics/drawable/NinePatchDrawable.smali	2013-03-18 17:26:30.000000000 +0100
***************
*** 209,214 ****
--- 209,220 ----
  
      iput v4, p0, Landroid/graphics/drawable/NinePatchDrawable;->mBitmapHeight:I
  
+     iget-object v4, p0, Landroid/graphics/drawable/NinePatchDrawable;->mNinePatchState:Landroid/graphics/drawable/NinePatchDrawable$NinePatchState;
+ 
+     iget-object v4, v4, Landroid/graphics/drawable/NinePatchDrawable$NinePatchState;->mPadding:Landroid/graphics/Rect;
+ 
+     iput-object v4, p0, Landroid/graphics/drawable/NinePatchDrawable;->mPadding:Landroid/graphics/Rect;
+ 
      :cond_0
      :goto_0
      return-void
