*** ./smali/android/content/res/Configuration.smali	2013-03-18 17:23:26.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/content/res/Configuration.smali	2013-03-18 17:26:18.000000000 +0100
***************
*** 142,147 ****
--- 142,149 ----
  
  .field public compatSmallestScreenWidthDp:I
  
+ .field public extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
  .field public fontScale:F
  
  .field public hardKeyboardHidden:I
***************
*** 180,185 ****
--- 182,194 ----
  
  .field public userSetLocale:Z
  
+ .field public static final UI_MODE_TYPE_SCALE_HUGE:I = 0xf
+ 
+ .field public static final UI_MODE_TYPE_SCALE_LARGE:I = 0xe
+ 
+ .field public static final UI_MODE_TYPE_SCALE_MEDIUM:I = 0xd
+ 
+ .field public static final UI_MODE_TYPE_SCALE_SMALL:I = 0xc
  
  # direct methods
  .method static constructor <clinit>()V
***************
*** 196,230 ****
  .end method
  
  .method public constructor <init>()V
!     .locals 0
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 0
      .parameter "o"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 0
      .parameter "source"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
--- 205,257 ----
  .end method
  
  .method public constructor <init>()V
!     .locals 1
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0}, Landroid/content/res/Configuration;->setToDefaults()V
  
      return-void
  .end method
  
  .method public constructor <init>(Landroid/content/res/Configuration;)V
!     .locals 1
      .parameter "o"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->setTo(Landroid/content/res/Configuration;)V
  
      return-void
  .end method
  
  .method private constructor <init>(Landroid/os/Parcel;)V
!     .locals 1
      .parameter "source"
  
      .prologue
      invoke-direct {p0}, Ljava/lang/Object;-><init>()V
  
+     new-instance v0, Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-direct {v0}, Lmiui/content/res/ExtraConfiguration;-><init>()V
+ 
+     iput-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
      invoke-virtual {p0, p1}, Landroid/content/res/Configuration;->readFromParcel(Landroid/os/Parcel;)V
  
      return-void
***************
*** 253,260 ****
--- 280,294 ----
  
      and-int/2addr v0, p0
  
+     if-nez v0, :cond_miui_add1
+ 
+     invoke-static {p0}, Lmiui/content/res/ExtraConfiguration;->needNewResources(I)Z
+ 
+     move-result v0
+ 
      if-eqz v0, :cond_0
  
+     :cond_miui_add1
      const/4 v0, 0x1
  
      :goto_0
***************
*** 491,496 ****
--- 525,538 ----
  
      sub-int v2, v3, v4
  
+     iget-object v3, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v4, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v3, v4}, Lmiui/content/res/ExtraConfiguration;->compareTo(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v2
+ 
      goto/16 :goto_0
  .end method
  
***************
*** 751,756 ****
--- 793,808 ----
      or-int/lit16 v0, v0, 0x800
  
      :cond_10
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v2, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1, v2}, Lmiui/content/res/ExtraConfiguration;->diff(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v1
+ 
+     or-int/2addr v0, v1
+ 
      return v0
  .end method
  
***************
*** 925,931 ****
  
      add-int v0, v1, v2
  
!     return v0
  
      :cond_0
      const/4 v1, 0x0
--- 977,991 ----
  
      add-int v0, v1, v2
  
!     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
! 
!     invoke-virtual {v1}, Lmiui/content/res/ExtraConfiguration;->hashCode()I
! 
!     move-result v1
! 
!     add-int/2addr v1, v0
! 
!     return v1
  
      :cond_0
      const/4 v1, 0x0
***************
*** 1180,1185 ****
--- 1240,1249 ----
  
      iput v0, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, p1}, Lmiui/content/res/ExtraConfiguration;->readFromParcel(Landroid/os/Parcel;)V
+ 
      return-void
  
      :cond_1
***************
*** 1189,1195 ****
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 1
      .parameter "o"
  
      .prologue
--- 1253,1259 ----
  .end method
  
  .method public setTo(Landroid/content/res/Configuration;)V
!     .locals 2
      .parameter "o"
  
      .prologue
***************
*** 1292,1297 ****
--- 1356,1367 ----
  
      iput v0, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v1, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, v1}, Lmiui/content/res/ExtraConfiguration;->setTo(Lmiui/content/res/ExtraConfiguration;)V
+ 
      return-void
  .end method
  
***************
*** 1349,1354 ****
--- 1419,1428 ----
  
      iput v1, p0, Landroid/content/res/Configuration;->seq:I
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0}, Lmiui/content/res/ExtraConfiguration;->setToDefaults()V
+ 
      return-void
  .end method
  
***************
*** 1643,1648 ****
--- 1717,1730 ----
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
  
      :cond_0
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1}, Lmiui/content/res/ExtraConfiguration;->toString()Ljava/lang/String;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
      const/16 v1, 0x7d
  
      invoke-virtual {v0, v1}, Ljava/lang/StringBuilder;->append(C)Ljava/lang/StringBuilder;
***************
*** 2450,2455 ****
--- 2532,2547 ----
      iput v1, p0, Landroid/content/res/Configuration;->seq:I
  
      :cond_17
+     iget-object v1, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     iget-object v2, p1, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v1, v2}, Lmiui/content/res/ExtraConfiguration;->updateFrom(Lmiui/content/res/ExtraConfiguration;)I
+ 
+     move-result v1
+ 
+     or-int/2addr v0, v1
+ 
      return v0
  
      :cond_18
***************
*** 2562,2567 ****
--- 2654,2663 ----
  
      invoke-virtual {p1, v0}, Landroid/os/Parcel;->writeInt(I)V
  
+     iget-object v0, p0, Landroid/content/res/Configuration;->extraConfig:Lmiui/content/res/ExtraConfiguration;
+ 
+     invoke-virtual {v0, p1, p2}, Lmiui/content/res/ExtraConfiguration;->writeToParcel(Landroid/os/Parcel;I)V
+ 
      return-void
  
      :cond_0
