*** ./smali/com/android/internal/telephony/AdnRecord.smali	2013-03-18 17:22:50.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/AdnRecord.smali	2013-03-18 17:25:38.000000000 +0100
***************
*** 247,252 ****
--- 247,254 ----
      and-int/lit16 v2, v3, 0xff
  
      .local v2, numberLength:I
+     goto :cond_0
+ 
      const/16 v3, 0xb
  
      if-le v2, v3, :cond_0
***************
*** 583,596 ****
  
      iget-object v5, p0, Lcom/android/internal/telephony/AdnRecord;->alphaTag:Ljava/lang/String;
  
!     invoke-static {v5}, Lcom/android/internal/telephony/GsmAlphabet;->stringToGsm8BitPacked(Ljava/lang/String;)[B
! 
!     move-result-object v2
! 
!     .local v2, byteTag:[B
!     array-length v5, v2
! 
!     invoke-static {v2, v9, v0, v9, v5}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
      goto :goto_1
  .end method
--- 585,591 ----
  
      iget-object v5, p0, Lcom/android/internal/telephony/AdnRecord;->alphaTag:Ljava/lang/String;
  
!     invoke-static {v0, v5, v3}, Lcom/android/internal/telephony/MiuiAdnUtils;->encodeAlphaTag([BLjava/lang/String;I)Z
  
      goto :goto_1
  .end method
