*** ./smali/com/android/internal/telephony/AdnRecordCache.smali	2013-03-18 17:22:51.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/AdnRecordCache.smali	2013-03-18 17:25:39.000000000 +0100
***************
*** 39,44 ****
--- 39,46 ----
      .end annotation
  .end field
  
+ .field private mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
  .field private mUsimPhoneBookManager:Lcom/android/internal/telephony/gsm/UsimPhoneBookManager;
  
  .field phone:Lcom/android/internal/telephony/PhoneBase;
***************
*** 89,94 ****
--- 91,102 ----
  
      iput-object v0, p0, Lcom/android/internal/telephony/AdnRecordCache;->mUsimPhoneBookManager:Lcom/android/internal/telephony/gsm/UsimPhoneBookManager;
  
+     new-instance v0, Lcom/android/internal/telephony/AdnCacheManager;
+ 
+     invoke-direct {v0, p0}, Lcom/android/internal/telephony/AdnCacheManager;-><init>(Lcom/android/internal/telephony/AdnRecordCache;)V
+ 
+     iput-object v0, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
      return-void
  .end method
  
***************
*** 380,392 ****
  
      if-nez v6, :cond_0
  
!     iget-object v7, p0, Lcom/android/internal/telephony/AdnRecordCache;->adnLikeFiles:Landroid/util/SparseArray;
  
!     iget-object v6, v1, Landroid/os/AsyncResult;->result:Ljava/lang/Object;
! 
!     check-cast v6, Ljava/util/ArrayList;
! 
!     invoke-virtual {v7, v2, v6}, Landroid/util/SparseArray;->put(ILjava/lang/Object;)V
  
      :cond_0
      invoke-direct {p0, v5, v1}, Lcom/android/internal/telephony/AdnRecordCache;->notifyWaiters(Ljava/util/ArrayList;Landroid/os/AsyncResult;)V
--- 388,396 ----
  
      if-nez v6, :cond_0
  
!     iget-object v6, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
  
!     invoke-virtual {v6, v2, v1}, Lcom/android/internal/telephony/AdnCacheManager;->handleLoadAllAdnLike(ILandroid/os/AsyncResult;)V
  
      :cond_0
      invoke-direct {p0, v5, v1}, Lcom/android/internal/telephony/AdnRecordCache;->notifyWaiters(Ljava/util/ArrayList;Landroid/os/AsyncResult;)V
***************
*** 491,496 ****
--- 495,504 ----
  
      if-eqz p3, :cond_0
  
+     iget-object v2, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
+     invoke-virtual {v2, p1, v0}, Lcom/android/internal/telephony/AdnCacheManager;->handleAllAdnLikeLoaded(ILjava/util/ArrayList;)V
+ 
      invoke-static {p3}, Landroid/os/AsyncResult;->forMessage(Landroid/os/Message;)Landroid/os/AsyncResult;
  
      move-result-object v2
***************
*** 613,618 ****
--- 621,630 ----
  
      invoke-direct {p0}, Lcom/android/internal/telephony/AdnRecordCache;->clearUserWriters()V
  
+     iget-object v0, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/AdnCacheManager;->reset()V
+ 
      return-void
  .end method
  
***************
*** 868,873 ****
--- 880,889 ----
  
      invoke-direct {p0, v0, v1}, Lcom/android/internal/telephony/AdnRecordCache;->sendErrorResponse(Landroid/os/Message;Ljava/lang/String;)V
  
+     iget-object v1, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
+     invoke-virtual {v1, p1}, Lcom/android/internal/telephony/AdnCacheManager;->handleNonExistentAdnRecord(I)V
+ 
      goto :goto_0
  
      :cond_4
***************
*** 909,914 ****
--- 925,936 ----
  
      .end local v9           #foundAdn:Lcom/android/internal/telephony/AdnRecord;
      :cond_6
+     iget-object v1, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
+     move-object/from16 v0, p3
+ 
+     invoke-virtual {v1, p1, p2, v0}, Lcom/android/internal/telephony/AdnCacheManager;->handleUpdateAdnRecord(ILcom/android/internal/telephony/AdnRecord;Lcom/android/internal/telephony/AdnRecord;)V
+ 
      iget-object v1, p0, Lcom/android/internal/telephony/AdnRecordCache;->userWriteResponse:Landroid/util/SparseArray;
  
      invoke-virtual {v1, p1}, Landroid/util/SparseArray;->get(I)Ljava/lang/Object;
***************
*** 975,977 ****
--- 997,1031 ----
  
      goto/16 :goto_0
  .end method
+ 
+ .method public getAdnCapacity()I
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/AdnCacheManager;->getAdnCapacity()I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
+ .method public getFreeAdn()I
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/AdnRecordCache;->mAdnCacheManager:Lcom/android/internal/telephony/AdnCacheManager;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/AdnCacheManager;->getFreeAdn()I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
\ No newline at end of file
