*** ./smali/com/android/internal/telephony/CallerInfoAsyncQuery.smali	2013-03-18 17:22:50.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/CallerInfoAsyncQuery.smali	2013-03-18 17:25:37.000000000 +0100
***************
*** 313,319 ****
      :cond_0
      sget-object v0, Landroid/provider/ContactsContract$PhoneLookup;->CONTENT_FILTER_URI:Landroid/net/Uri;
  
!     invoke-static {p2}, Landroid/net/Uri;->encode(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
--- 313,323 ----
      :cond_0
      sget-object v0, Landroid/provider/ContactsContract$PhoneLookup;->CONTENT_FILTER_URI:Landroid/net/Uri;
  
!     invoke-static {p2}, Lcom/android/internal/telephony/CallerInfoAsyncQuery;->getPhoneNumber(Ljava/lang/String;)Ljava/lang/String;
! 
!     move-result-object v1
! 
!     invoke-static {v1}, Landroid/net/Uri;->encode(Ljava/lang/String;)Ljava/lang/String;
  
      move-result-object v1
  
***************
*** 391,393 ****
--- 395,423 ----
  
      return-void
  .end method
+ 
+ .method private static getPhoneNumber(Ljava/lang/String;)Ljava/lang/String;
+     .locals 2
+     .parameter "number"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     invoke-static {p0}, Lmiui/telephony/PhoneNumberUtils$PhoneNumber;->parse(Ljava/lang/CharSequence;)Lmiui/telephony/PhoneNumberUtils$PhoneNumber;
+ 
+     move-result-object v0
+ 
+     .local v0, pn:Lmiui/telephony/PhoneNumberUtils$PhoneNumber;
+     if-eqz v0, :cond_0
+ 
+     const/4 v1, 0x1
+ 
+     invoke-virtual {v0, v1}, Lmiui/telephony/PhoneNumberUtils$PhoneNumber;->getNumberWithoutPrefix(Z)Ljava/lang/String;
+ 
+     move-result-object p0
+ 
+     .end local p0
+     :cond_0
+     return-object p0
+ .end method
\ No newline at end of file
