*** ./smali/com/android/internal/telephony/CallerInfo.smali	2013-03-18 17:22:49.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/CallerInfo.smali	2013-03-18 17:25:36.000000000 +0100
***************
*** 26,31 ****
--- 26,33 ----
  
  .field public contactRingtoneUri:Landroid/net/Uri;
  
+ .field public extra:Lmiui/telephony/ExtraCallerInfo;
+ 
  .field public geoDescription:Ljava/lang/String;
  
  .field public isCachedPhotoCurrent:Z
***************
*** 89,94 ****
--- 91,102 ----
  
      iput-boolean v0, p0, Lcom/android/internal/telephony/CallerInfo;->mIsVoiceMail:Z
  
+     new-instance v0, Lmiui/telephony/ExtraCallerInfo;
+ 
+     invoke-direct {v0}, Lmiui/telephony/ExtraCallerInfo;-><init>()V
+ 
+     iput-object v0, p0, Lcom/android/internal/telephony/CallerInfo;->extra:Lmiui/telephony/ExtraCallerInfo;
+ 
      return-void
  .end method
  
***************
*** 280,286 ****
      :cond_1
      const-string v3, "number"
  
!     invoke-interface {p2, v3}, Landroid/database/Cursor;->getColumnIndex(Ljava/lang/String;)I
  
      move-result v0
  
--- 288,294 ----
      :cond_1
      const-string v3, "number"
  
!     invoke-static {p1, v3, p2}, Lmiui/telephony/CallerInfo;->getColumnIndex(Landroid/net/Uri;Ljava/lang/String;Landroid/database/Cursor;)I
  
      move-result v0
  
***************
*** 295,301 ****
      :cond_2
      const-string v3, "normalized_number"
  
!     invoke-interface {p2, v3}, Landroid/database/Cursor;->getColumnIndex(Ljava/lang/String;)I
  
      move-result v0
  
--- 303,309 ----
      :cond_2
      const-string v3, "normalized_number"
  
!     invoke-static {p1, v3, p2}, Lmiui/telephony/CallerInfo;->getColumnIndex(Landroid/net/Uri;Ljava/lang/String;Landroid/database/Cursor;)I
  
      move-result v0
  
***************
*** 310,316 ****
      :cond_3
      const-string v3, "label"
  
!     invoke-interface {p2, v3}, Landroid/database/Cursor;->getColumnIndex(Ljava/lang/String;)I
  
      move-result v0
  
--- 318,324 ----
      :cond_3
      const-string v3, "label"
  
!     invoke-static {p1, v3, p2}, Lmiui/telephony/CallerInfo;->getColumnIndex(Landroid/net/Uri;Ljava/lang/String;Landroid/database/Cursor;)I
  
      move-result v0
  
***************
*** 318,324 ****
  
      const-string v3, "type"
  
!     invoke-interface {p2, v3}, Landroid/database/Cursor;->getColumnIndex(Ljava/lang/String;)I
  
      move-result v2
  
--- 326,332 ----
  
      const-string v3, "type"
  
!     invoke-static {p1, v3, p2}, Lmiui/telephony/CallerInfo;->getColumnIndex(Landroid/net/Uri;Ljava/lang/String;Landroid/database/Cursor;)I
  
      move-result v2
  
***************
*** 441,446 ****
--- 449,460 ----
  
      iput-boolean v4, v1, Lcom/android/internal/telephony/CallerInfo;->contactExists:Z
  
+     invoke-static {p0, v1, p2}, Lmiui/telephony/ExtraCallerInfo;->getExtraCallerInfo(Landroid/content/Context;Lcom/android/internal/telephony/CallerInfo;Landroid/database/Cursor;)Lmiui/telephony/ExtraCallerInfo;
+ 
+     move-result-object v3
+ 
+     iput-object v3, v1, Lcom/android/internal/telephony/CallerInfo;->extra:Lmiui/telephony/ExtraCallerInfo;
+ 
      .end local v0           #columnIndex:I
      :cond_6
      invoke-interface {p2}, Landroid/database/Cursor;->close()V
