*** ./smali/com/android/internal/telephony/DataConnectionTracker.smali	2013-03-18 17:22:50.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/DataConnectionTracker.smali	2013-03-18 17:25:38.000000000 +0100
***************
*** 1412,1417 ****
--- 1412,1424 ----
      monitor-enter v2
  
      :try_start_0
+     invoke-virtual {p0}, Lcom/android/internal/telephony/DataConnectionTracker;->isMmsDataEnabled()Z
+ 
+     move-result v0
+ 
+     .local v0, result:Z
+     if-nez v0, :cond_miui_add1
+ 
      iget-boolean v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mInternalDataEnabled:Z
  
      if-eqz v1, :cond_1
***************
*** 1431,1436 ****
--- 1438,1444 ----
      const/4 v0, 0x1
  
      .local v0, result:Z
+     :cond_miui_add1
      :goto_0
      monitor-exit v2
      :try_end_0
***************
*** 3374,3376 ****
--- 3382,3430 ----
  
  .method protected abstract stopNetStatPoll()V
  .end method
+ 
+ .method protected isMmsDataEnabled()Z
+     .locals 3
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x1
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mRequestedApnType:Ljava/lang/String;
+ 
+     const-string v2, "mms"
+ 
+     invoke-virtual {v1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/DataConnectionTracker;->mPhone:Lcom/android/internal/telephony/PhoneBase;
+ 
+     invoke-virtual {v1}, Lcom/android/internal/telephony/PhoneBase;->getContext()Landroid/content/Context;
+ 
+     move-result-object v1
+ 
+     invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
+ 
+     move-result-object v1
+ 
+     const-string v2, "always_enable_mms"
+ 
+     invoke-static {v1, v2, v0}, Landroid/provider/Settings$System;->getInt(Landroid/content/ContentResolver;Ljava/lang/String;I)I
+ 
+     move-result v1
+ 
+     if-ne v1, v0, :cond_0
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
\ No newline at end of file
