*** ./smali/com/android/internal/telephony/IccPhoneBookInterfaceManager.smali	2013-03-18 17:22:50.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/IccPhoneBookInterfaceManager.smali	2013-03-18 17:25:37.000000000 +0100
***************
*** 677,679 ****
--- 677,705 ----
      :cond_0
      return-void
  .end method
+ 
+ .method public getAdnCapacity()I
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/IccPhoneBookInterfaceManager;->adnCache:Lcom/android/internal/telephony/AdnRecordCache;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/AdnRecordCache;->getAdnCapacity()I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
+ 
+ .method public getFreeAdn()I
+     .locals 1
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/telephony/IccPhoneBookInterfaceManager;->adnCache:Lcom/android/internal/telephony/AdnRecordCache;
+ 
+     invoke-virtual {v0}, Lcom/android/internal/telephony/AdnRecordCache;->getFreeAdn()I
+ 
+     move-result v0
+ 
+     return v0
+ .end method
