*** ./smali/com/android/internal/telephony/SmsUsageMonitor.smali	2013-03-18 17:22:48.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/SmsUsageMonitor.smali	2013-03-18 17:25:35.000000000 +0100
***************
*** 317,322 ****
--- 317,335 ----
      .parameter "smsWaiting"
  
      .prologue
+     const-string v1, "com.android.mms"
+ 
+     invoke-virtual {v1, p1}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_miui_add1
+ 
+     const/4 v1, 0x1
+ 
+     return v1
+ 
+     :cond_miui_add1
      iget-object v2, p0, Lcom/android/internal/telephony/SmsUsageMonitor;->mSmsStamp:Ljava/util/HashMap;
  
      monitor-enter v2
