*** ./smali/com/android/server/pm/Settings.smali	2013-03-18 17:22:37.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/services.jar.out/./smali/com/android/server/pm/Settings.smali	2013-03-18 17:25:21.000000000 +0100
***************
*** 2516,2524 ****
--- 2516,2536 ----
      move-result v3
  
      iput v3, v2, Lcom/android/server/pm/PackageSettingBase;->enabled:I
+ 
+     iget v3, v2, Lcom/android/server/pm/PackageSettingBase;->enabled:I
+ 
+     const/4 v5, 0x2
+ 
+     if-ne v3, v5, :cond_miui_add1
+ 
+     const/4 v3, 0x0
+ 
+     iput v3, v2, Lcom/android/server/pm/PackageSettingBase;->enabled:I
+ 
      :try_end_b
      .catch Ljava/lang/NumberFormatException; {:try_start_b .. :try_end_b} :catch_1
  
+     :cond_miui_add1
      :goto_7
      const/4 v3, 0x0
  
