*** ./smali/com/android/internal/telephony/cdma/CdmaServiceStateTracker.smali	2013-03-18 17:22:50.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/cdma/CdmaServiceStateTracker.smali	2013-03-18 17:25:37.000000000 +0100
***************
*** 6,12 ****
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$3;
      }
  .end annotation
  
--- 6,13 ----
  # annotations
  .annotation system Ldalvik/annotation/MemberClasses;
      value = {
!         Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$3;,
!         Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$LocaleChangedIntentReceiver;
      }
  .end annotation
  
***************
*** 106,111 ****
--- 107,114 ----
  
  .field phone:Lcom/android/internal/telephony/cdma/CDMAPhone;
  
+ .field private mIntentReceiver:Landroid/content/BroadcastReceiver;
+ 
  
  # direct methods
  .method public constructor <init>(Lcom/android/internal/telephony/cdma/CDMAPhone;)V
***************
*** 183,188 ****
--- 186,197 ----
  
      iput-object v6, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->currentCarrier:Ljava/lang/String;
  
+     new-instance v4, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$LocaleChangedIntentReceiver;
+ 
+     invoke-direct {v4, p0, v6}, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$LocaleChangedIntentReceiver;-><init>(Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$1;)V
+ 
+     iput-object v4, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->mIntentReceiver:Landroid/content/BroadcastReceiver;
+ 
      new-instance v4, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker$1;
  
      new-instance v5, Landroid/os/Handler;
***************
*** 364,369 ****
--- 373,380 ----
  
      iput-boolean v3, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->mNeedToRegForRuimLoaded:Z
  
+     invoke-direct {p0}, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->monitorLocaleChange()V
+ 
      return-void
  .end method
  
***************
*** 6978,6980 ****
--- 6989,7020 ----
  
      goto :goto_0
  .end method
+ 
+ .method private monitorLocaleChange()V
+     .locals 3
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     new-instance v0, Landroid/content/IntentFilter;
+ 
+     invoke-direct {v0}, Landroid/content/IntentFilter;-><init>()V
+ 
+     .local v0, filter:Landroid/content/IntentFilter;
+     const-string v1, "android.intent.action.LOCALE_CHANGED"
+ 
+     invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
+ 
+     iget-object v1, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->phone:Lcom/android/internal/telephony/cdma/CDMAPhone;
+ 
+     invoke-virtual {v1}, Lcom/android/internal/telephony/cdma/CDMAPhone;->getContext()Landroid/content/Context;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p0, Lcom/android/internal/telephony/cdma/CdmaServiceStateTracker;->mIntentReceiver:Landroid/content/BroadcastReceiver;
+ 
+     invoke-virtual {v1, v2, v0}, Landroid/content/Context;->registerReceiver(Landroid/content/BroadcastReceiver;Landroid/content/IntentFilter;)Landroid/content/Intent;
+ 
+     return-void
+ .end method
