*** ./smali/com/android/internal/widget/LockPatternUtils$PasswordFileObserver.smali	2013-03-18 17:22:46.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/LockPatternUtils$PasswordFileObserver.smali	2013-03-18 17:25:32.000000000 +0100
***************
*** 93,99 ****
  
      move-result v2
  
!     if-eqz v2, :cond_0
  
      const-string v2, "LockPatternUtils"
  
--- 93,99 ----
  
      move-result v2
  
!     if-eqz v2, :cond_miui_add1
  
      const-string v2, "LockPatternUtils"
  
***************
*** 130,133 ****
--- 130,174 ----
      move v0, v1
  
      goto :goto_2
+ 
+     :cond_miui_add1
+     const-string v2, "access_control.key"
+ 
+     invoke-virtual {v2, p2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
+ 
+     move-result v2
+ 
+     if-eqz v2, :cond_0
+ 
+     const-string v2, "LockPatternUtils"
+ 
+     const-string v3, "access control password file changed"
+ 
+     invoke-static {v2, v3}, Landroid/util/Log;->d(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     sget-object v2, Lmiui/security/MiuiLockPatternUtils;->sHaveNonZeroACFile:Ljava/util/concurrent/atomic/AtomicBoolean;
+ 
+     new-instance v3, Ljava/io/File;
+ 
+     sget-object v4, Lmiui/security/MiuiLockPatternUtils;->sLockACFilename:Ljava/lang/String;
+ 
+     invoke-direct {v3, v4}, Ljava/io/File;-><init>(Ljava/lang/String;)V
+ 
+     invoke-virtual {v3}, Ljava/io/File;->length()J
+ 
+     move-result-wide v3
+ 
+     cmp-long v3, v3, v5
+ 
+     if-lez v3, :cond_miui_add2
+ 
+     :goto_miui_add1
+     invoke-virtual {v2, v0}, Ljava/util/concurrent/atomic/AtomicBoolean;->set(Z)V
+ 
+     goto :goto_1
+ 
+     :cond_miui_add2
+     move v0, v1
+ 
+     goto :goto_miui_add1
  .end method
