*** ./smali/com/android/internal/widget/ScrollingTabContainerView.smali	2013-03-18 17:22:47.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/ScrollingTabContainerView.smali	2013-03-18 17:25:33.000000000 +0100
***************
*** 184,190 ****
      .locals 4
  
      .prologue
!     new-instance v0, Landroid/widget/LinearLayout;
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ScrollingTabContainerView;->getContext()Landroid/content/Context;
  
--- 184,190 ----
      .locals 4
  
      .prologue
!     new-instance v0, Lcom/android/internal/widget/MiuiTabLayout;
  
      invoke-virtual {p0}, Lcom/android/internal/widget/ScrollingTabContainerView;->getContext()Landroid/content/Context;
  
***************
*** 194,200 ****
  
      const v3, 0x10102f4
  
!     invoke-direct {v0, v1, v2, v3}, Landroid/widget/LinearLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
      .local v0, tabLayout:Landroid/widget/LinearLayout;
      const/4 v1, 0x1
--- 194,200 ----
  
      const v3, 0x10102f4
  
!     invoke-direct {v0, v1, v2, v3}, Lcom/android/internal/widget/MiuiTabLayout;-><init>(Landroid/content/Context;Landroid/util/AttributeSet;I)V
  
      .local v0, tabLayout:Landroid/widget/LinearLayout;
      const/4 v1, 0x1
