*** ./smali/com/android/internal/view/menu/MenuItemImpl.smali	2013-03-18 17:22:47.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/view/menu/MenuItemImpl.smali	2013-03-18 17:25:34.000000000 +0100
***************
*** 1662,1664 ****
--- 1662,1690 ----
  
      return-object v0
  .end method
+ 
+ .method public isForceShowText()Z
+     .locals 2
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget v0, p0, Lcom/android/internal/view/menu/MenuItemImpl;->mShowAsAction:I
+ 
+     const/high16 v1, -0x8000
+ 
+     and-int/2addr v0, v1
+ 
+     if-eqz v0, :cond_0
+ 
+     const/4 v0, 0x1
+ 
+     :goto_0
+     return v0
+ 
+     :cond_0
+     const/4 v0, 0x0
+ 
+     goto :goto_0
+ .end method
\ No newline at end of file
