*** ./smali/com/android/server/am/AppErrorDialog$1.smali	2013-03-18 17:22:39.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/services.jar.out/./smali/com/android/server/am/AppErrorDialog$1.smali	2013-03-18 17:25:23.000000000 +0100
***************
*** 34,43 ****
  
  # virtual methods
  .method public handleMessage(Landroid/os/Message;)V
!     .locals 3
      .parameter "msg"
  
      .prologue
      iget-object v0, p0, Lcom/android/server/am/AppErrorDialog$1;->this$0:Lcom/android/server/am/AppErrorDialog;
  
      #getter for: Lcom/android/server/am/AppErrorDialog;->mProc:Lcom/android/server/am/ProcessRecord;
--- 34,92 ----
  
  # virtual methods
  .method public handleMessage(Landroid/os/Message;)V
!     .locals 5
      .parameter "msg"
  
      .prologue
+ 
+     const/4 v0, 0x1
+ 
+     iget-object v1, p0, Lcom/android/server/am/AppErrorDialog$1;->this$0:Lcom/android/server/am/AppErrorDialog;
+ 
+     #getter for: Lcom/android/server/am/AppErrorDialog;->mProc:Lcom/android/server/am/ProcessRecord;
+     invoke-static {v1}, Lcom/android/server/am/AppErrorDialog;->access$000(Lcom/android/server/am/AppErrorDialog;)Lcom/android/server/am/ProcessRecord;
+ 
+     move-result-object v1
+ 
+     if-eqz v1, :cond_miui_add1
+ 
+     iget-object v1, p0, Lcom/android/server/am/AppErrorDialog$1;->this$0:Lcom/android/server/am/AppErrorDialog;
+ 
+     #getter for: Lcom/android/server/am/AppErrorDialog;->mCrashInfo:Landroid/app/ApplicationErrorReport$CrashInfo;
+     invoke-static {v1}, Lcom/android/server/am/AppErrorDialog;->access$200(Lcom/android/server/am/AppErrorDialog;)Landroid/app/ApplicationErrorReport$CrashInfo;
+ 
+     move-result-object v1
+ 
+     if-eqz v1, :cond_miui_add1
+ 
+     iget-object v1, p0, Lcom/android/server/am/AppErrorDialog$1;->this$0:Lcom/android/server/am/AppErrorDialog;
+ 
+     invoke-virtual {v1}, Lcom/android/server/am/AppErrorDialog;->getContext()Landroid/content/Context;
+ 
+     move-result-object v1
+ 
+     iget-object v2, p0, Lcom/android/server/am/AppErrorDialog$1;->this$0:Lcom/android/server/am/AppErrorDialog;
+ 
+     #getter for: Lcom/android/server/am/AppErrorDialog;->mProc:Lcom/android/server/am/ProcessRecord;
+     invoke-static {v2}, Lcom/android/server/am/AppErrorDialog;->access$000(Lcom/android/server/am/AppErrorDialog;)Lcom/android/server/am/ProcessRecord;
+ 
+     move-result-object v2
+ 
+     iget-object v3, p0, Lcom/android/server/am/AppErrorDialog$1;->this$0:Lcom/android/server/am/AppErrorDialog;
+ 
+     #getter for: Lcom/android/server/am/AppErrorDialog;->mCrashInfo:Landroid/app/ApplicationErrorReport$CrashInfo;
+     invoke-static {v3}, Lcom/android/server/am/AppErrorDialog;->access$200(Lcom/android/server/am/AppErrorDialog;)Landroid/app/ApplicationErrorReport$CrashInfo;
+ 
+     move-result-object v3
+ 
+     iget v4, p1, Landroid/os/Message;->what:I
+ 
+     if-ne v4, v0, :cond_miui_add2
+ 
+     :goto_miui_add1
+     invoke-static {v1, v2, v3, v0}, Lcom/android/server/am/MiuiErrorReport;->sendFcErrorReport(Landroid/content/Context;Lcom/android/server/am/ProcessRecord;Landroid/app/ApplicationErrorReport$CrashInfo;Z)V
+ 
+     :cond_miui_add1
      iget-object v0, p0, Lcom/android/server/am/AppErrorDialog$1;->this$0:Lcom/android/server/am/AppErrorDialog;
  
      #getter for: Lcom/android/server/am/AppErrorDialog;->mProc:Lcom/android/server/am/ProcessRecord;
***************
*** 103,108 ****
--- 152,162 ----
  
      return-void
  
+     :cond_miui_add2
+     const/4 v0, 0x0
+ 
+     goto :goto_miui_add1
+ 
      :catchall_0
      move-exception v0
  
