*** ./smali/android/net/Proxy.smali	2013-03-18 17:23:40.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/net/Proxy.smali	2013-03-18 17:26:27.000000000 +0100
***************
*** 254,260 ****
  .end method
  
  .method public static final getPreferredHttpHost(Landroid/content/Context;Ljava/lang/String;)Lorg/apache/http/HttpHost;
!     .locals 6
      .parameter "context"
      .parameter "url"
  
--- 254,260 ----
  .end method
  
  .method public static final getPreferredHttpHost(Landroid/content/Context;Ljava/lang/String;)Lorg/apache/http/HttpHost;
!     .locals 3
      .parameter "context"
      .parameter "url"
  
***************
*** 285,303 ****
      check-cast v1, Ljava/net/InetSocketAddress;
  
      .local v1, sa:Ljava/net/InetSocketAddress;
-     new-instance v2, Lorg/apache/http/HttpHost;
  
!     invoke-virtual {v1}, Ljava/net/InetSocketAddress;->getHostName()Ljava/lang/String;
  
!     move-result-object v3
! 
!     invoke-virtual {v1}, Ljava/net/InetSocketAddress;->getPort()I
! 
!     move-result v4
! 
!     const-string v5, "http"
! 
!     invoke-direct {v2, v3, v4, v5}, Lorg/apache/http/HttpHost;-><init>(Ljava/lang/String;ILjava/lang/String;)V
  
      goto :goto_0
  .end method
--- 285,294 ----
      check-cast v1, Ljava/net/InetSocketAddress;
  
      .local v1, sa:Ljava/net/InetSocketAddress;
  
!     invoke-static {v1}, Landroid/net/ExtraProxy;->getPreferredHttpHost(Ljava/net/InetSocketAddress;)Lorg/apache/http/HttpHost;
  
!     move-result-object v2
  
      goto :goto_0
  .end method
