*** ./smali/android/content/res/Resources.smali	2013-03-18 17:23:25.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/content/res/Resources.smali	2013-03-18 17:26:18.000000000 +0100
***************
*** 189,195 ****
      return-void
  .end method
  
! .method private constructor <init>()V
      .locals 3
  
      .prologue
--- 189,195 ----
      return-void
  .end method
  
! .method constructor <init>()V
      .locals 3
  
      .prologue
***************
*** 765,773 ****
      .restart local v0       #attrs:Landroid/content/res/TypedArray;
      :cond_1
      :try_start_1
-     new-instance v0, Landroid/content/res/TypedArray;
- 
-     .end local v0           #attrs:Landroid/content/res/TypedArray;
      mul-int/lit8 v2, p1, 0x6
  
      new-array v2, v2, [I
--- 765,770 ----
***************
*** 776,783 ****
  
      new-array v4, v4, [I
  
!     invoke-direct {v0, p0, v2, v4, p1}, Landroid/content/res/TypedArray;-><init>(Landroid/content/res/Resources;[I[II)V
  
      monitor-exit v3
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
--- 773,783 ----
  
      new-array v4, v4, [I
  
!     invoke-static {p0, v2, v4, p1}, Landroid/content/res/MiuiClassFactory;->newTypedArray(Landroid/content/res/Resources;[I[II)Landroid/content/res/TypedArray;
  
+     move-result-object v0
+ 
+     .end local v0           #attrs:Landroid/content/res/TypedArray; 
      monitor-exit v3
      :try_end_1
      .catchall {:try_start_1 .. :try_end_1} :catchall_0
***************
*** 839,850 ****
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     new-instance v0, Landroid/content/res/Resources;
  
!     .end local v0           #ret:Landroid/content/res/Resources;
!     invoke-direct {v0}, Landroid/content/res/Resources;-><init>()V
  
-     .restart local v0       #ret:Landroid/content/res/Resources;
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
      :cond_0
--- 839,848 ----
      .local v0, ret:Landroid/content/res/Resources;
      if-nez v0, :cond_0
  
!     invoke-static {}, Landroid/content/res/MiuiClassFactory;->newResources()Landroid/content/res/Resources;
  
!     move-result-object v0
  
      sput-object v0, Landroid/content/res/Resources;->mSystem:Landroid/content/res/Resources;
  
      :cond_0
***************
*** 3649,3654 ****
--- 3647,3658 ----
      .end local v5           #e:Ljava/lang/Exception;
      .end local v11           #rnf:Landroid/content/res/Resources$NotFoundException;
      :cond_9
+     invoke-virtual/range {p0 .. p2}, Landroid/content/res/Resources;->loadOverlayDrawable(Landroid/util/TypedValue;I)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v3
+ 
+     if-nez v3, :cond_3
+ 
      :try_start_1
      move-object/from16 v0, p0
  
***************
*** 4137,4143 ****
      throw v7
  .end method
  
! .method public final newTheme()Landroid/content/res/Resources$Theme;
      .locals 1
  
      .prologue
--- 4141,4147 ----
      throw v7
  .end method
  
! .method public newTheme()Landroid/content/res/Resources$Theme;
      .locals 1
  
      .prologue
***************
*** 4993,4999 ****
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v20
  
      :cond_5
      move-object/from16 v0, p0
--- 4997,5009 ----
  
      invoke-static/range {v20 .. v20}, Landroid/content/pm/ActivityInfo;->activityInfoConfigToNative(I)I
  
!     move-result v2
! 
!     const/high16 v3, -0x8000
! 
!     and-int v3, v3, v20
! 
!     or-int v20, v2, v3
  
      :cond_5
      move-object/from16 v0, p0
***************
*** 5369,5371 ****
--- 5379,5417 ----
  
      throw v2
  .end method
+ 
+ .method static clearPreloadedCache()V
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->mPreloadedColorStateLists:Landroid/util/SparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/SparseArray;->clear()V
+ 
+     sget-object v0, Landroid/content/res/Resources;->sPreloadedColorDrawables:Landroid/util/LongSparseArray;
+ 
+     invoke-virtual {v0}, Landroid/util/LongSparseArray;->clear()V
+ 
+     return-void
+ .end method
+ 
+ .method loadOverlayDrawable(Landroid/util/TypedValue;I)Landroid/graphics/drawable/Drawable;
+     .locals 1
+     .parameter "value"
+     .parameter "id"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     const/4 v0, 0x0
+ 
+     return-object v0
+ .end method
