*** ./smali/android/widget/Toast.smali	2013-03-18 17:23:04.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/widget/Toast.smali	2013-03-18 17:25:56.000000000 +0100
***************
*** 429,431 ****
--- 429,451 ----
  
      goto :goto_0
  .end method
+ 
+ .method public setType(I)V
+     .locals 1
+     .parameter "type"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/widget/Toast;->mTN:Landroid/widget/Toast$TN;
+ 
+     #getter for: Landroid/widget/Toast$TN;->mParams:Landroid/view/WindowManager$LayoutParams;
+     invoke-static {v0}, Landroid/widget/Toast$TN;->access$000(Landroid/widget/Toast$TN;)Landroid/view/WindowManager$LayoutParams;
+ 
+     move-result-object v0
+ 
+     iput p1, v0, Landroid/view/WindowManager$LayoutParams;->type:I
+ 
+     return-void
+ .end method
\ No newline at end of file
