*** ./smali/android/content/pm/ApplicationInfo.smali	2013-03-18 17:23:22.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/content/pm/ApplicationInfo.smali	2013-03-18 17:26:15.000000000 +0100
***************
*** 26,31 ****
--- 26,33 ----
      .end annotation
  .end field
  
+ .field public static final FLAG_ACCESS_CONTROL_PASSWORD:I = -0x80000000
+ 
  .field public static final FLAG_ALLOW_BACKUP:I = 0x8000
  
  .field public static final FLAG_ALLOW_CLEAR_USER_DATA:I = 0x40
***************
*** 36,41 ****
--- 38,45 ----
  
  .field public static final FLAG_DEBUGGABLE:I = 0x2
  
+ .field public static final FLAG_DISABLE_AUTOSTART:I = 0x40000000
+ 
  .field public static final FLAG_EXTERNAL_STORAGE:I = 0x40000
  
  .field public static final FLAG_FACTORY_TEST:I = 0x10
