*** ./smali/com/android/server/net/NetworkPolicyManagerService.smali	2013-03-18 17:22:41.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/services.jar.out/./smali/com/android/server/net/NetworkPolicyManagerService.smali	2013-03-18 17:25:26.000000000 +0100
***************
*** 741,747 ****
  .end method
  
  .method private static buildViewDataUsageIntent(Landroid/net/NetworkTemplate;)Landroid/content/Intent;
!     .locals 4
      .parameter "template"
  
      .prologue
--- 741,747 ----
  .end method
  
  .method private static buildViewDataUsageIntent(Landroid/net/NetworkTemplate;)Landroid/content/Intent;
!     .locals 2
      .parameter "template"
  
      .prologue
***************
*** 750,764 ****
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
      .local v0, intent:Landroid/content/Intent;
!     new-instance v1, Landroid/content/ComponentName;
! 
!     const-string v2, "com.android.settings"
! 
!     const-string v3, "com.android.settings.Settings$DataUsageSummaryActivity"
  
!     invoke-direct {v1, v2, v3}, Landroid/content/ComponentName;-><init>(Ljava/lang/String;Ljava/lang/String;)V
! 
!     invoke-virtual {v0, v1}, Landroid/content/Intent;->setComponent(Landroid/content/ComponentName;)Landroid/content/Intent;
  
      const/high16 v1, 0x1000
  
--- 750,758 ----
      invoke-direct {v0}, Landroid/content/Intent;-><init>()V
  
      .local v0, intent:Landroid/content/Intent;
!     const-string v1, "android.intent.action.VIEW_DATA_USAGE_SUMMARY"
  
!     invoke-virtual {v0, v1}, Landroid/content/Intent;->setAction(Ljava/lang/String;)Landroid/content/Intent;
  
      const/high16 v1, 0x1000
  
***************
*** 1035,1041 ****
      return-void
  .end method
  
! .method private enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
      .locals 14
      .parameter "policy"
      .parameter "type"
--- 1029,1035 ----
      return-void
  .end method
  
! .method protected enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
      .locals 14
      .parameter "policy"
      .parameter "type"
***************
*** 1058,1066 ****
  
      invoke-virtual {v7, v0}, Landroid/app/Notification$Builder;->setOnlyAlertOnce(Z)Landroid/app/Notification$Builder;
  
-     const/4 v0, 0x1
- 
-     invoke-virtual {v7, v0}, Landroid/app/Notification$Builder;->setOngoing(Z)Landroid/app/Notification$Builder;
  
      iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mContext:Landroid/content/Context;
  
--- 1052,1057 ----
***************
*** 1876,1887 ****
  .end method
  
  .method private getTotalBytes(Landroid/net/NetworkTemplate;JJ)J
!     .locals 7
      .parameter "template"
      .parameter "start"
      .parameter "end"
  
      .prologue
      :try_start_0
      iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mNetworkStats:Landroid/net/INetworkStatsService;
  
--- 1867,1881 ----
  .end method
  
  .method private getTotalBytes(Landroid/net/NetworkTemplate;JJ)J
!     .locals 12
      .parameter "template"
      .parameter "start"
      .parameter "end"
  
      .prologue
+     const-wide/16 v10, 0x0
+ 
+     .local v10, total:J
      :try_start_0
      iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mNetworkStats:Landroid/net/INetworkStatsService;
  
***************
*** 1889,1895 ****
  
      move-wide v2, p2
  
!     move-wide v4, p4
  
      invoke-interface/range {v0 .. v5}, Landroid/net/INetworkStatsService;->getSummaryForNetwork(Landroid/net/NetworkTemplate;JJ)Landroid/net/NetworkStats;
  
--- 1883,1889 ----
  
      move-wide v2, p2
  
!     move-wide/from16 v4, p4
  
      invoke-interface/range {v0 .. v5}, Landroid/net/INetworkStatsService;->getSummaryForNetwork(Landroid/net/NetworkTemplate;JJ)Landroid/net/NetworkStats;
  
***************
*** 1899,1915 ****
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
!     move-result-wide v0
  
      :goto_0
!     return-wide v0
  
!     :catch_0
!     move-exception v6
  
!     .local v6, e:Landroid/os/RemoteException;
      const-wide/16 v0, 0x0
  
      goto :goto_0
  .end method
  
--- 1893,1963 ----
      :try_end_0
      .catch Landroid/os/RemoteException; {:try_start_0 .. :try_end_0} :catch_0
  
!     move-result-wide v10
  
      :goto_0
!     invoke-virtual {p1}, Landroid/net/NetworkTemplate;->getMatchRule()I
  
!     move-result v0
! 
!     const/4 v1, 0x1
! 
!     if-ne v0, v1, :cond_miui_add1
! 
!     iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mContext:Landroid/content/Context;
! 
!     invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
! 
!     move-result-object v0
! 
!     const-string v1, "data_usage_adjusting_time"
! 
!     const-wide/16 v2, 0x0
! 
!     invoke-static {v0, v1, v2, v3}, Landroid/provider/Settings$Secure;->getLong(Landroid/content/ContentResolver;Ljava/lang/String;J)J
  
!     move-result-wide v6
! 
!     .local v6, adjustingTime:J
!     cmp-long v0, v6, p2
! 
!     if-ltz v0, :cond_miui_add1
! 
!     cmp-long v0, v6, p4
! 
!     if-gtz v0, :cond_miui_add1
! 
!     iget-object v0, p0, Lcom/android/server/net/NetworkPolicyManagerService;->mContext:Landroid/content/Context;
! 
!     invoke-virtual {v0}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
! 
!     move-result-object v0
! 
!     const-string v1, "data_usage_adjustment"
! 
!     const-wide/16 v2, 0x0
! 
!     invoke-static {v0, v1, v2, v3}, Landroid/provider/Settings$Secure;->getLong(Landroid/content/ContentResolver;Ljava/lang/String;J)J
! 
!     move-result-wide v8
! 
!     .local v8, adjustment:J
      const-wide/16 v0, 0x0
  
+     add-long v2, v10, v8
+ 
+     invoke-static {v0, v1, v2, v3}, Ljava/lang/Math;->max(JJ)J
+ 
+     move-result-wide v10
+ 
+     .end local v6           #adjustingTime:J
+     .end local v8           #adjustment:J
+     :cond_miui_add1
+     return-wide v10
+ 
+     :catch_0
+     move-exception v0
+ 
      goto :goto_0
  .end method
  
***************
*** 2497,2503 ****
      goto :goto_0
  .end method
  
! .method private setNetworkTemplateEnabled(Landroid/net/NetworkTemplate;Z)V
      .locals 2
      .parameter "template"
      .parameter "enabled"
--- 2546,2552 ----
      goto :goto_0
  .end method
  
! .method protected setNetworkTemplateEnabled(Landroid/net/NetworkTemplate;Z)V
      .locals 2
      .parameter "template"
      .parameter "enabled"
***************
*** 2721,2727 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v1, v2}, Lcom/android/server/net/NetworkPolicyManagerService;->setNetworkTemplateEnabled(Landroid/net/NetworkTemplate;Z)V
  
      goto :goto_0
  
--- 2770,2776 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v1, v2}, Lcom/android/server/net/NetworkPolicyManagerService;->setNetworkTemplateEnabled(Landroid/net/NetworkTemplate;Z)V
  
      goto :goto_0
  
***************
*** 2769,2775 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v1, v9}, Lcom/android/server/net/NetworkPolicyManagerService;->setNetworkTemplateEnabled(Landroid/net/NetworkTemplate;Z)V
  
      goto :goto_0
  
--- 2818,2824 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v1, v9}, Lcom/android/server/net/NetworkPolicyManagerService;->setNetworkTemplateEnabled(Landroid/net/NetworkTemplate;Z)V
  
      goto :goto_0
  
***************
*** 3470,3476 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v11, v1, v13, v14}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto :goto_0
  
--- 3519,3525 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v11, v1, v13, v14}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto :goto_0
  
***************
*** 3479,3485 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v11, v1, v13, v14}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      iget-object v1, v11, Landroid/net/NetworkPolicy;->template:Landroid/net/NetworkTemplate;
  
--- 3528,3534 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v11, v1, v13, v14}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      iget-object v1, v11, Landroid/net/NetworkPolicy;->template:Landroid/net/NetworkTemplate;
  
***************
*** 3514,3520 ****
  
      move-object/from16 v0, p0
  
!     invoke-direct {v0, v11, v1, v13, v14}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto :goto_0
  
--- 3563,3569 ----
  
      move-object/from16 v0, p0
  
!     invoke-virtual {v0, v11, v1, v13, v14}, Lcom/android/server/net/NetworkPolicyManagerService;->enqueueNotification(Landroid/net/NetworkPolicy;IJ)V
  
      goto :goto_0
  
