*** ./smali/com/android/internal/widget/ActionBarContainer.smali	2013-03-18 17:22:46.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/widget/ActionBarContainer.smali	2013-03-18 17:25:32.000000000 +0100
***************
*** 499,505 ****
  
      if-eqz v8, :cond_5
  
!     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mStackedBackground:Landroid/graphics/drawable/Drawable;
  
      iget-object v9, p0, Lcom/android/internal/widget/ActionBarContainer;->mTabContainer:Landroid/view/View;
  
--- 499,517 ----
  
      if-eqz v8, :cond_5
  
!     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mBackground:Landroid/graphics/drawable/Drawable;
! 
!     if-eqz v8, :cond_miui_add1
! 
!     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mActionBarView:Lcom/android/internal/widget/ActionBarView;
! 
!     invoke-virtual {v8}, Lcom/android/internal/widget/ActionBarView;->getHeight()I
! 
!     move-result v8
! 
!     if-nez v8, :cond_miui_add1
! 
!     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mBackground:Landroid/graphics/drawable/Drawable;
  
      iget-object v9, p0, Lcom/android/internal/widget/ActionBarContainer;->mTabContainer:Landroid/view/View;
  
***************
*** 527,532 ****
--- 539,549 ----
  
      invoke-virtual {v8, v9, v10, v11, v12}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
  
+     const/4 v8, 0x0
+ 
+     iput-boolean v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mIsStacked:Z
+ 
+     :goto_6
      const/4 v6, 0x1
  
      goto :goto_4
***************
*** 535,540 ****
--- 552,588 ----
      const/4 v8, 0x0
  
      goto :goto_5
+ 
+     :cond_miui_add1
+     iget-object v8, p0, Lcom/android/internal/widget/ActionBarContainer;->mStackedBackground:Landroid/graphics/drawable/Drawable;
+ 
+     iget-object v9, p0, Lcom/android/internal/widget/ActionBarContainer;->mTabContainer:Landroid/view/View;
+ 
+     invoke-virtual {v9}, Landroid/view/View;->getLeft()I
+ 
+     move-result v9
+ 
+     iget-object v10, p0, Lcom/android/internal/widget/ActionBarContainer;->mTabContainer:Landroid/view/View;
+ 
+     invoke-virtual {v10}, Landroid/view/View;->getTop()I
+ 
+     move-result v10
+ 
+     iget-object v11, p0, Lcom/android/internal/widget/ActionBarContainer;->mTabContainer:Landroid/view/View;
+ 
+     invoke-virtual {v11}, Landroid/view/View;->getRight()I
+ 
+     move-result v11
+ 
+     iget-object v12, p0, Lcom/android/internal/widget/ActionBarContainer;->mTabContainer:Landroid/view/View;
+ 
+     invoke-virtual {v12}, Landroid/view/View;->getBottom()I
+ 
+     move-result v12
+ 
+     invoke-virtual {v8, v9, v10, v11, v12}, Landroid/graphics/drawable/Drawable;->setBounds(IIII)V
+ 
+     goto :goto_6
  .end method
  
  .method public onMeasure(II)V
