*** ./smali/com/android/internal/telephony/IIccPhoneBook$Stub.smali	2013-03-18 17:22:50.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/IIccPhoneBook$Stub.smali	2013-03-18 17:25:38.000000000 +0100
***************
*** 34,39 ****
--- 34,42 ----
  
  .field static final TRANSACTION_updateAdnRecordsInEfBySearch:I = 0x2
  
+ .field static final TRANSACTION_getFreeAdn:I = 0x5
+ 
+ .field static final TRANSACTION_getAdnCapacity:I = 0x6
  
  # direct methods
  .method public constructor <init>()V
***************
*** 293,298 ****
--- 296,336 ----
  
      goto/16 :goto_0
  
+     .end local v1           #_arg0:I
+     .end local v7           #_result:[I
+     :sswitch_miui_add1
+     const-string v0, "com.android.internal.telephony.IIccPhoneBook"
+ 
+     invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual {p0}, Lcom/android/internal/telephony/IIccPhoneBook$Stub;->getFreeAdn()I
+ 
+     move-result v7
+ 
+     .local v7, _result:I
+     invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     goto/16 :goto_0
+ 
+     .end local v7           #_result:I
+     :sswitch_miui_add2
+     const-string v0, "com.android.internal.telephony.IIccPhoneBook"
+ 
+     invoke-virtual {p2, v0}, Landroid/os/Parcel;->enforceInterface(Ljava/lang/String;)V
+ 
+     invoke-virtual {p0}, Lcom/android/internal/telephony/IIccPhoneBook$Stub;->getAdnCapacity()I
+ 
+     move-result v7
+ 
+     .restart local v7       #_result:I
+     invoke-virtual {p3}, Landroid/os/Parcel;->writeNoException()V
+ 
+     invoke-virtual {p3, v7}, Landroid/os/Parcel;->writeInt(I)V
+ 
+     goto/16 :goto_0
+ 
      nop
  
      :sswitch_data_0
***************
*** 301,306 ****
--- 339,346 ----
          0x2 -> :sswitch_2
          0x3 -> :sswitch_3
          0x4 -> :sswitch_4
+         0x5 -> :sswitch_miui_add1
+         0x6 -> :sswitch_miui_add2
          0x5f4e5446 -> :sswitch_0
      .end sparse-switch
  .end method
