*** ./smali/com/android/internal/telephony/SMSDispatcher.smali	2013-03-18 17:22:52.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/SMSDispatcher.smali	2013-03-18 17:25:40.000000000 +0100
***************
*** 849,855 ****
  
      move-result-object v1
  
!     invoke-virtual {v0, v1}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([B)I
  
      move-result v8
  
--- 849,859 ----
  
      move-result-object v1
  
!     invoke-virtual {p1}, Lcom/android/internal/telephony/SmsMessageBase;->getOriginatingAddress()Ljava/lang/String;
! 
!     move-result-object v2
! 
!     invoke-virtual {v0, v1, v2}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([BLjava/lang/String;)I
  
      move-result v8
  
***************
*** 913,922 ****
  .end method
  
  .method protected dispatchPdus([[B)V
!     .locals 3
      .parameter "pdus"
  
      .prologue
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_RECEIVED"
--- 917,945 ----
  .end method
  
  .method protected dispatchPdus([[B)V
!     .locals 4
      .parameter "pdus"
  
      .prologue
+     iget-object v1, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v1, p1}, Lmiui/provider/ExtraTelephony;->checkFirewallForSms(Landroid/content/Context;[[B)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_miui_add1
+ 
+     const/4 v1, 0x1
+ 
+     const/4 v2, -0x1
+ 
+     const/4 v3, 0x0
+ 
+     invoke-virtual {p0, v1, v2, v3}, Lcom/android/internal/telephony/SMSDispatcher;->acknowledgeLastIncomingSms(ZILandroid/os/Message;)V
+ 
+     goto :goto_miui_add1
+ 
+     :cond_miui_add1
      new-instance v0, Landroid/content/Intent;
  
      const-string v1, "android.provider.Telephony.SMS_RECEIVED"
***************
*** 940,954 ****
  
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/telephony/SMSDispatcher;->dispatch(Landroid/content/Intent;Ljava/lang/String;)V
  
      return-void
  .end method
  
  .method protected dispatchPortAddressedPdus([[BI)V
!     .locals 4
      .parameter "pdus"
      .parameter "port"
  
      .prologue
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
--- 963,997 ----
  
      invoke-virtual {p0, v0, v1}, Lcom/android/internal/telephony/SMSDispatcher;->dispatch(Landroid/content/Intent;Ljava/lang/String;)V
  
+     :goto_miui_add1
      return-void
  .end method
  
  .method protected dispatchPortAddressedPdus([[BI)V
!     .locals 5
      .parameter "pdus"
      .parameter "port"
  
      .prologue
+     iget-object v2, p0, Lcom/android/internal/telephony/SMSDispatcher;->mContext:Landroid/content/Context;
+ 
+     invoke-static {v2, p1}, Lmiui/provider/ExtraTelephony;->checkFirewallForSms(Landroid/content/Context;[[B)Z
+ 
+     move-result v2
+ 
+     if-eqz v2, :cond_miui_add1
+ 
+     const/4 v2, 0x1
+ 
+     const/4 v3, -0x1
+ 
+     const/4 v4, 0x0
+ 
+     invoke-virtual {p0, v2, v3, v4}, Lcom/android/internal/telephony/SMSDispatcher;->acknowledgeLastIncomingSms(ZILandroid/os/Message;)V
+ 
+     goto :goto_miui_add1
+ 
+     :cond_miui_add1
      new-instance v2, Ljava/lang/StringBuilder;
  
      invoke-direct {v2}, Ljava/lang/StringBuilder;-><init>()V
***************
*** 995,1000 ****
--- 1038,1044 ----
  
      invoke-virtual {p0, v0, v2}, Lcom/android/internal/telephony/SMSDispatcher;->dispatch(Landroid/content/Intent;Ljava/lang/String;)V
  
+     :goto_miui_add1
      return-void
  .end method
  
***************
*** 2252,2258 ****
  
      iget-object v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mWapPush:Lcom/android/internal/telephony/WapPushOverSms;
  
!     invoke-virtual {v3, v13}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([B)I
  
      move-result v3
  
--- 2296,2304 ----
  
      iget-object v3, v0, Lcom/android/internal/telephony/SMSDispatcher;->mWapPush:Lcom/android/internal/telephony/WapPushOverSms;
  
!     move-object/from16 v0, p2
! 
!     invoke-virtual {v3, v13, v0}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([BLjava/lang/String;)I
  
      move-result v3
  
***************
*** 2347,2353 ****
  
      move-result-object v4
  
!     invoke-virtual {v3, v4}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([B)I
  
      move-result v3
  
--- 2393,2401 ----
  
      move-result-object v4
  
!     move-object/from16 v0, p2
! 
!     invoke-virtual {v3, v4, v0}, Lcom/android/internal/telephony/WapPushOverSms;->dispatchWapPdu([BLjava/lang/String;)I
  
      move-result v3
  
