*** ./smali/com/android/internal/telephony/WapPushOverSms.smali	2013-03-18 17:22:48.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/com/android/internal/telephony/WapPushOverSms.smali	2013-03-18 17:25:35.000000000 +0100
***************
*** 75,82 ****
  
  
  # virtual methods
! .method public dispatchWapPdu([B)I
!     .locals 24
      .parameter "pdu"
  
      .prologue
--- 75,82 ----
  
  
  # virtual methods
! .method public dispatchWapPdu([BLjava/lang/String;)I
!     .locals 26
      .parameter "pdu"
  
      .prologue
***************
*** 425,431 ****
  
      invoke-static {v0, v6, v13, v1, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
!     goto :goto_1
  
      .end local v6           #dataIndex:I
      .restart local v20       #wapAppId:Ljava/lang/String;
--- 425,459 ----
  
      invoke-static {v0, v6, v13, v1, v2}, Ljava/lang/System;->arraycopy(Ljava/lang/Object;ILjava/lang/Object;II)V
  
!     move-object/from16 v0, p0
! 
!     iget-object v0, v0, Lcom/android/internal/telephony/WapPushOverSms;->mContext:Landroid/content/Context;
! 
!     move-object/from16 v22, v0
! 
!     move-object/from16 v0, v22
! 
!     invoke-static {v0, v13}, Lmiui/provider/ExtraTelephony;->checkFirewallForWapPush(Landroid/content/Context;[B)Z
! 
!     move-result v22
! 
!     if-eqz v22, :goto_1
! 
!     move-object/from16 v0, p0
! 
!     iget-object v0, v0, Lcom/android/internal/telephony/WapPushOverSms;->mSmsDispatcher:Lcom/android/internal/telephony/SMSDispatcher;
! 
!     move-object/from16 v20, v0
! 
!     const/16 v21, 0x1
! 
!     const/16 v22, -0x1
! 
!     const/16 v23, 0x0
! 
!     invoke-virtual/range {v20 .. v23}, Lcom/android/internal/telephony/SMSDispatcher;->acknowledgeLastIncomingSms(ZILandroid/os/Message;)V
! 
!     goto :goto_0
  
      .end local v6           #dataIndex:I
      .restart local v20       #wapAppId:Ljava/lang/String;
***************
*** 599,604 ****
--- 627,640 ----
  
      invoke-virtual {v12, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/io/Serializable;)Landroid/content/Intent;
  
+     const-string v22, "address"
+ 
+     move-object/from16 v0, v22
+ 
+     move-object/from16 v1, p2
+ 
+     invoke-virtual {v12, v0, v1}, Landroid/content/Intent;->putExtra(Ljava/lang/String;Ljava/lang/String;)Landroid/content/Intent;
+ 
      move-object/from16 v0, p0
  
      iget-object v0, v0, Lcom/android/internal/telephony/WapPushOverSms;->mSmsDispatcher:Lcom/android/internal/telephony/SMSDispatcher;
