*** ./smali/com/android/internal/policy/impl/LockPatternKeyguardView.smali	2013-03-18 17:23:55.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/LockPatternKeyguardView.smali	2013-03-18 17:26:43.000000000 +0100
***************
*** 80,88 ****
  
  .field private final mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
! .field private mLockScreen:Landroid/view/View;
  
! .field private mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
  
  .field private mRecreateRunnable:Ljava/lang/Runnable;
  
--- 80,88 ----
  
  .field private final mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
! .field mLockScreen:Landroid/view/View;
  
! .field mMode:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
  
  .field private mRecreateRunnable:Ljava/lang/Runnable;
  
***************
*** 90,98 ****
  
  .field private mSavedState:Landroid/os/Parcelable;
  
! .field private volatile mScreenOn:Z
  
! .field private mShowLockBeforeUnlock:Z
  
  .field private mTransportControlView:Lcom/android/internal/widget/TransportControlView;
  
--- 90,98 ----
  
  .field private mSavedState:Landroid/os/Parcelable;
  
! .field volatile mScreenOn:Z
  
! .field mShowLockBeforeUnlock:Z
  
  .field private mTransportControlView:Lcom/android/internal/widget/TransportControlView;
  
***************
*** 242,250 ****
  
      invoke-virtual {v0, p0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->registerInfoCallback(Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallback;)V
  
!     new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;
  
!     invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;-><init>(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mKeyguardScreenCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
--- 242,250 ----
  
      invoke-virtual {v0, p0}, Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;->registerInfoCallback(Lcom/android/internal/policy/impl/KeyguardUpdateMonitor$InfoCallback;)V
  
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
!     move-result-object v0
  
      iput-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mKeyguardScreenCallback:Lcom/android/internal/policy/impl/KeyguardScreenCallback;
  
***************
*** 254,260 ****
  
      invoke-virtual {p0, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->setDescendantFocusability(I)V
  
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
  
      move-result-object v0
  
--- 254,260 ----
  
      invoke-virtual {p0, v0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->setDescendantFocusability(I)V
  
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
  
      move-result-object v0
  
***************
*** 580,586 ****
      .parameter "x0"
  
      .prologue
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
  
      move-result v0
  
--- 580,586 ----
      .parameter "x0"
  
      .prologue
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
  
      move-result v0
  
***************
*** 742,748 ****
      goto :goto_2
  .end method
  
! .method private getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
      .locals 2
  
      .prologue
--- 742,748 ----
      goto :goto_2
  .end method
  
! .method getInitialMode()Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
      .locals 2
  
      .prologue
***************
*** 753,759 ****
      move-result-object v0
  
      .local v0, simState:Lcom/android/internal/telephony/IccCard$State;
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
  
      move-result v1
  
--- 753,759 ----
      move-result-object v0
  
      .local v0, simState:Lcom/android/internal/telephony/IccCard$State;
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->stuckOnLockScreenBecauseSimMissing()Z
  
      move-result v1
  
***************
*** 1161,1167 ****
      return-void
  .end method
  
! .method private recreateLockScreen()V
      .locals 2
  
      .prologue
--- 1161,1167 ----
      return-void
  .end method
  
! .method recreateLockScreen()V
      .locals 2
  
      .prologue
***************
*** 1631,1637 ****
      goto :goto_0
  .end method
  
! .method private stuckOnLockScreenBecauseSimMissing()Z
      .locals 2
  
      .prologue
--- 1631,1637 ----
      goto :goto_0
  .end method
  
! .method stuckOnLockScreenBecauseSimMissing()Z
      .locals 2
  
      .prologue
***************
*** 1703,1709 ****
      if-nez v4, :cond_2
  
      :cond_1
!     invoke-direct {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->recreateLockScreen()V
  
      :cond_2
      sget-object v4, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;->UnlockScreen:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
--- 1703,1709 ----
      if-nez v4, :cond_2
  
      :cond_1
!     invoke-virtual {p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->recreateLockScreen()V
  
      :cond_2
      sget-object v4, Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;->UnlockScreen:Lcom/android/internal/policy/impl/LockPatternKeyguardView$Mode;
***************
*** 2163,2169 ****
      if-ne p1, v3, :cond_3
  
      :try_start_0
!     new-instance v2, Lcom/android/internal/policy/impl/AccountUnlockScreen;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
--- 2163,2169 ----
      if-ne p1, v3, :cond_3
  
      :try_start_0
!     new-instance v2, Lcom/android/internal/policy/impl/MiuiAccountUnlockScreen;
  
      iget-object v3, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mContext:Landroid/content/Context;
  
***************
*** 2175,2181 ****
  
      iget-object v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v2 .. v7}, Lcom/android/internal/policy/impl/AccountUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
      :try_end_0
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_0
  
--- 2175,2181 ----
  
      iget-object v7, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
  
!     invoke-direct/range {v2 .. v7}, Lcom/android/internal/policy/impl/MiuiAccountUnlockScreen;-><init>(Landroid/content/Context;Landroid/content/res/Configuration;Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;Lcom/android/internal/policy/impl/KeyguardScreenCallback;Lcom/android/internal/widget/LockPatternUtils;)V
      :try_end_0
      .catch Ljava/lang/IllegalStateException; {:try_start_0 .. :try_end_0} :catch_0
  
***************
*** 3070,3072 ****
--- 3070,3122 ----
  
      goto :goto_0
  .end method
+ 
+ .method protected createKeyguardScreenCallback()Lcom/android/internal/policy/impl/KeyguardScreenCallback;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     new-instance v0, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;
+ 
+     invoke-direct {v0, p0}, Lcom/android/internal/policy/impl/LockPatternKeyguardView$3;-><init>(Lcom/android/internal/policy/impl/LockPatternKeyguardView;)V
+ 
+     return-object v0
+ .end method
+ 
+ .method getConfiguration()Landroid/content/res/Configuration;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mConfiguration:Landroid/content/res/Configuration;
+ 
+     return-object v0
+ .end method
+ 
+ .method getLockPatternUtils()Lcom/android/internal/widget/LockPatternUtils;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mLockPatternUtils:Lcom/android/internal/widget/LockPatternUtils;
+ 
+     return-object v0
+ .end method
+ 
+ .method getUpdateMonitor()Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+     .locals 1
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Lcom/android/internal/policy/impl/LockPatternKeyguardView;->mUpdateMonitor:Lcom/android/internal/policy/impl/KeyguardUpdateMonitor;
+ 
+     return-object v0
+ .end method
