*** ./smali/com/android/internal/policy/impl/PhoneWindowManager.smali	2013-03-18 17:23:56.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/PhoneWindowManager.smali	2013-03-18 17:26:44.000000000 +0100
***************
*** 236,242 ****
  
  .field mForceStatusBar:Z
  
! .field mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
  .field private mHDMIObserver:Landroid/os/UEventObserver;
  
--- 236,242 ----
  
  .field mForceStatusBar:Z
  
! .field mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
  .field private mHDMIObserver:Landroid/os/UEventObserver;
  
***************
*** 322,328 ****
  
  .field private mPowerKeyTriggered:Z
  
! .field private final mPowerLongPress:Ljava/lang/Runnable;
  
  .field mPowerManager:Landroid/os/LocalPowerManager;
  
--- 322,328 ----
  
  .field private mPowerKeyTriggered:Z
  
! .field mPowerLongPress:Ljava/lang/Runnable;
  
  .field mPowerManager:Landroid/os/LocalPowerManager;
  
***************
*** 352,358 ****
  
  .field mScreenOnFully:Z
  
! .field private final mScreenshotChordLongPress:Ljava/lang/Runnable;
  
  .field mScreenshotConnection:Landroid/content/ServiceConnection;
  
--- 352,358 ----
  
  .field mScreenOnFully:Z
  
! .field final mScreenshotChordLongPress:Ljava/lang/Runnable;
  
  .field mScreenshotConnection:Landroid/content/ServiceConnection;
  
***************
*** 413,419 ****
  
  .field mVirtualKeyVibePattern:[J
  
! .field private mVolumeDownKeyConsumedByScreenshotChord:Z
  
  .field private mVolumeDownKeyTime:J
  
--- 413,419 ----
  
  .field mVirtualKeyVibePattern:[J
  
! .field mVolumeDownKeyConsumedByScreenshotChord:Z
  
  .field private mVolumeDownKeyTime:J
  
***************
*** 1038,1044 ****
      return-void
  .end method
  
! .method private interceptPowerKeyUp(Z)Z
      .locals 3
      .parameter "canceled"
  
--- 1038,1044 ----
      return-void
  .end method
  
! .method interceptPowerKeyUp(Z)Z
      .locals 3
      .parameter "canceled"
  
***************
*** 4846,4854 ****
  
      iput-object p4, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mPowerManager:Landroid/os/LocalPowerManager;
  
!     new-instance v4, Lcom/android/internal/policy/impl/KeyguardViewMediator;
  
!     invoke-direct {v4, p1, p0, p4}, Lcom/android/internal/policy/impl/KeyguardViewMediator;-><init>(Landroid/content/Context;Lcom/android/internal/policy/impl/PhoneWindowManager;Landroid/os/LocalPowerManager;)V
  
      iput-object v4, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/KeyguardViewMediator;
  
--- 4846,4854 ----
  
      iput-object p4, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mPowerManager:Landroid/os/LocalPowerManager;
  
!     invoke-static {p1, p0, p4}, Lcom/android/internal/policy/impl/MiuiClassFactory;->createKeyguardViewMediator(Landroid/content/Context;Lcom/android/internal/policy/impl/PhoneWindowManager;Landroid/os/LocalPowerManager;)Lcom/android/internal/policy/impl/KeyguardViewMediator;
  
!     move-result-object v4
  
      iput-object v4, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/KeyguardViewMediator;
  
***************
*** 6621,6628 ****
  
      .restart local v7       #keyguardActive:Z
      :cond_4
-     if-eqz v1, :cond_5
- 
      and-int/lit16 v10, p2, 0x100
  
      if-eqz v10, :cond_5
--- 6621,6626 ----
***************
*** 6635,6642 ****
--- 6633,6648 ----
  
      const/4 v10, 0x0
  
+     if-eqz v1, :cond_miui_add1
+ 
      const/4 v11, 0x1
  
+     goto :goto_miui_add1
+ 
+     :cond_miui_add1
+     const/4 v11, 0x2
+ 
+     :goto_miui_add1
      const/4 v12, 0x0
  
      invoke-virtual {p0, v10, v11, v12}, Lcom/android/internal/policy/impl/PhoneWindowManager;->performHapticFeedbackLw(Landroid/view/WindowManagerPolicy$WindowState;IZ)Z
***************
*** 6966,6972 ****
      .end local v3           #hungUp:Z
      .end local v9           #telephonyService:Lcom/android/internal/telephony/ITelephony;
      :cond_15
!     invoke-direct {p0, v0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->interceptPowerKeyUp(Z)Z
  
      move-result v10
  
--- 6972,6978 ----
      .end local v3           #hungUp:Z
      .end local v9           #telephonyService:Lcom/android/internal/telephony/ITelephony;
      :cond_15
!     invoke-virtual {p0, v0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->interceptPowerKeyUp(Z)Z
  
      move-result v10
  
***************
*** 7131,7137 ****
      const/4 v10, 0x1
  
      :goto_c
!     invoke-direct {p0, v10}, Lcom/android/internal/policy/impl/PhoneWindowManager;->interceptPowerKeyUp(Z)Z
  
      move-result v10
  
--- 7137,7143 ----
      const/4 v10, 0x1
  
      :goto_c
!     invoke-virtual {p0, v10}, Lcom/android/internal/policy/impl/PhoneWindowManager;->interceptPowerKeyUp(Z)Z
  
      move-result v10
  
***************
*** 7146,7151 ****
--- 7152,7171 ----
  
      iput-boolean v10, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mPendingPowerKeyUpCanceled:Z
  
+     iget-object v10, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
+ 
+     if-eqz v10, :cond_0
+ 
+     iget-object v10, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
+ 
+     new-instance v11, Landroid/content/Intent;
+ 
+     const-string v12, "android.intent.action.KEYCODE_POWER_UP"
+ 
+     invoke-direct {v11, v12}, Landroid/content/Intent;-><init>(Ljava/lang/String;)V
+ 
+     invoke-virtual {v10, v11}, Landroid/content/Context;->sendBroadcast(Landroid/content/Intent;)V
+ 
      goto/16 :goto_3
  
      :cond_1f
***************
*** 10349,10365 ****
      .locals 3
  
      .prologue
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
      if-nez v1, :cond_0
  
!     new-instance v1, Lcom/android/internal/policy/impl/GlobalActions;
  
      iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
  
!     invoke-direct {v1, v2}, Lcom/android/internal/policy/impl/GlobalActions;-><init>(Landroid/content/Context;)V
  
!     iput-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/KeyguardViewMediator;
--- 10369,10385 ----
      .locals 3
  
      .prologue
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      if-nez v1, :cond_0
  
!     new-instance v1, Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      iget-object v2, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mContext:Landroid/content/Context;
  
!     invoke-direct {v1, v2}, Lcom/android/internal/policy/impl/MiuiGlobalActions;-><init>(Landroid/content/Context;)V
  
!     iput-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      :cond_0
      iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mKeyguardMediator:Lcom/android/internal/policy/impl/KeyguardViewMediator;
***************
*** 10369,10381 ****
      move-result v0
  
      .local v0, keyguardShowing:Z
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/GlobalActions;
  
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->isDeviceProvisioned()Z
  
      move-result v2
  
!     invoke-virtual {v1, v0, v2}, Lcom/android/internal/policy/impl/GlobalActions;->showDialog(ZZ)V
  
      if-eqz v0, :cond_1
  
--- 10389,10401 ----
      move-result v0
  
      .local v0, keyguardShowing:Z
!     iget-object v1, p0, Lcom/android/internal/policy/impl/PhoneWindowManager;->mGlobalActions:Lcom/android/internal/policy/impl/MiuiGlobalActions;
  
      invoke-virtual {p0}, Lcom/android/internal/policy/impl/PhoneWindowManager;->isDeviceProvisioned()Z
  
      move-result v2
  
!     invoke-virtual {v1, v0, v2}, Lcom/android/internal/policy/impl/MiuiGlobalActions;->showDialog(ZZ)V
  
      if-eqz v0, :cond_1
  
