*** ./smali/com/android/internal/policy/impl/Policy.smali	2013-03-18 17:23:55.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/android.policy.jar.out/./smali/com/android/internal/policy/impl/Policy.smali	2013-03-18 17:26:43.000000000 +0100
***************
*** 138,148 ****
  .method public makeNewFallbackEventHandler(Landroid/content/Context;)Landroid/view/FallbackEventHandler;
      .locals 1
      .parameter "context"
  
      .prologue
!     new-instance v0, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;
  
!     invoke-direct {v0, p1}, Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;-><init>(Landroid/content/Context;)V
  
      return-object v0
  .end method
--- 138,152 ----
  .method public makeNewFallbackEventHandler(Landroid/content/Context;)Landroid/view/FallbackEventHandler;
      .locals 1
      .parameter "context"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->CHANGE_CODE:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
  
      .prologue
!     invoke-static {p1}, Lcom/android/internal/policy/impl/MiuiClassFactory;->createPhoneFallbackEventHandler(Landroid/content/Context;)Lcom/android/internal/policy/impl/PhoneFallbackEventHandler;
  
!     move-result-object v0
  
      return-object v0
  .end method
***************
*** 175,183 ****
      .locals 1
  
      .prologue
!     new-instance v0, Lcom/android/internal/policy/impl/PhoneWindowManager;
  
!     invoke-direct {v0}, Lcom/android/internal/policy/impl/PhoneWindowManager;-><init>()V
  
      return-object v0
  .end method
--- 179,187 ----
      .locals 1
  
      .prologue
!     invoke-static {}, Lcom/android/internal/policy/impl/MiuiClassFactory;->createPhoneWindowManager()Lcom/android/internal/policy/impl/PhoneWindowManager;
  
!     move-result-object v0
  
      return-object v0
  .end method
