*** ./smali/com/android/server/BackupManagerService.smali	2013-03-18 17:22:40.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/services.jar.out/./smali/com/android/server/BackupManagerService.smali	2013-03-18 17:25:23.000000000 +0100
***************
*** 7361,7367 ****
  
      const-string v2, "fullback"
  
!     invoke-virtual {p0, v11, v2}, Lcom/android/server/BackupManagerService;->startConfirmationUi(ILjava/lang/String;)Z
  
      move-result v2
  
--- 7361,7367 ----
  
      const-string v2, "fullback"
  
!     invoke-virtual {p0, v11, v2}, Lcom/android/server/BackupManagerService;->startBackupRestore(ILjava/lang/String;)Z
  
      move-result v2
  
***************
*** 7441,7447 ****
  
      invoke-virtual {v2, v3, v4, v5}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-virtual {p0, v11, v1}, Lcom/android/server/BackupManagerService;->startConfirmationTimeout(ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v2, "BackupManagerService"
  
--- 7441,7447 ----
  
      invoke-virtual {v2, v3, v4, v5}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-virtual {p0, v11, v1}, Lcom/android/server/BackupManagerService;->doNothing(ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v2, "BackupManagerService"
  
***************
*** 7600,7606 ****
  
      const-string v5, "fullrest"
  
!     invoke-virtual {p0, v4, v5}, Lcom/android/server/BackupManagerService;->startConfirmationUi(ILjava/lang/String;)Z
  
      move-result v5
  
--- 7600,7606 ----
  
      const-string v5, "fullrest"
  
!     invoke-virtual {p0, v4, v5}, Lcom/android/server/BackupManagerService;->startBackupRestore(ILjava/lang/String;)Z
  
      move-result v5
  
***************
*** 7680,7686 ****
  
      invoke-virtual {v5, v6, v7, v8}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-virtual {p0, v4, v3}, Lcom/android/server/BackupManagerService;->startConfirmationTimeout(ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v5, "BackupManagerService"
  
--- 7680,7686 ----
  
      invoke-virtual {v5, v6, v7, v8}, Landroid/os/PowerManager;->userActivity(JZ)V
  
!     invoke-virtual {p0, v4, v3}, Lcom/android/server/BackupManagerService;->doNothing(ILcom/android/server/BackupManagerService$FullParams;)V
  
      const-string v5, "BackupManagerService"
  
***************
*** 10857,10859 ****
--- 10857,11001 ----
  
      goto :goto_0
  .end method
+ 
+ .method startBackupRestore(ILjava/lang/String;)Z
+     .locals 7
+     .parameter "token"
+     .parameter "action"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v4, p0, Lcom/android/server/BackupManagerService;->mFullConfirmations:Landroid/util/SparseArray;
+ 
+     monitor-enter v4
+ 
+     :try_start_0
+     iget-object v3, p0, Lcom/android/server/BackupManagerService;->mFullConfirmations:Landroid/util/SparseArray;
+ 
+     invoke-virtual {v3, p1}, Landroid/util/SparseArray;->get(I)Ljava/lang/Object;
+ 
+     move-result-object v1
+ 
+     check-cast v1, Lcom/android/server/BackupManagerService$FullParams;
+ 
+     .local v1, params:Lcom/android/server/BackupManagerService$FullParams;
+     if-eqz v1, :cond_1
+ 
+     iget-object v3, p0, Lcom/android/server/BackupManagerService;->mFullConfirmations:Landroid/util/SparseArray;
+ 
+     invoke-virtual {v3, p1}, Landroid/util/SparseArray;->delete(I)V
+ 
+     instance-of v3, v1, Lcom/android/server/BackupManagerService$FullBackupParams;
+ 
+     if-eqz v3, :cond_0
+ 
+     const/4 v2, 0x2
+ 
+     .local v2, verb:I
+     :goto_0
+     const/4 v3, 0x0
+ 
+     iput-object v3, v1, Lcom/android/server/BackupManagerService$FullParams;->observer:Landroid/app/backup/IFullBackupRestoreObserver;
+ 
+     const-string v3, ""
+ 
+     iput-object v3, v1, Lcom/android/server/BackupManagerService$FullParams;->curPassword:Ljava/lang/String;
+ 
+     iget-object v3, p0, Lcom/android/server/BackupManagerService;->mContext:Landroid/content/Context;
+ 
+     invoke-virtual {v3}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
+ 
+     move-result-object v3
+ 
+     sget-object v5, Lmiui/provider/ExtraSettings$Secure;->APP_ENCRYPT_PASSWORD:Ljava/lang/String;
+ 
+     invoke-static {v3, v5}, Landroid/provider/Settings$Secure;->getString(Landroid/content/ContentResolver;Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v3
+ 
+     iput-object v3, v1, Lcom/android/server/BackupManagerService$FullParams;->encryptPassword:Ljava/lang/String;
+ 
+     const-string v3, "BackupManagerService"
+ 
+     new-instance v5, Ljava/lang/StringBuilder;
+ 
+     invoke-direct {v5}, Ljava/lang/StringBuilder;-><init>()V
+ 
+     const-string v6, "Sending conf message with verb "
+ 
+     invoke-virtual {v5, v6}, Ljava/lang/StringBuilder;->append(Ljava/lang/String;)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5, v2}, Ljava/lang/StringBuilder;->append(I)Ljava/lang/StringBuilder;
+ 
+     move-result-object v5
+ 
+     invoke-virtual {v5}, Ljava/lang/StringBuilder;->toString()Ljava/lang/String;
+ 
+     move-result-object v5
+ 
+     invoke-static {v3, v5}, Landroid/util/Slog;->d(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     iget-object v3, p0, Lcom/android/server/BackupManagerService;->mWakelock:Landroid/os/PowerManager$WakeLock;
+ 
+     invoke-virtual {v3}, Landroid/os/PowerManager$WakeLock;->acquire()V
+ 
+     iget-object v3, p0, Lcom/android/server/BackupManagerService;->mBackupHandler:Lcom/android/server/BackupManagerService$BackupHandler;
+ 
+     invoke-virtual {v3, v2, v1}, Lcom/android/server/BackupManagerService$BackupHandler;->obtainMessage(ILjava/lang/Object;)Landroid/os/Message;
+ 
+     move-result-object v0
+ 
+     .local v0, msg:Landroid/os/Message;
+     iget-object v3, p0, Lcom/android/server/BackupManagerService;->mBackupHandler:Lcom/android/server/BackupManagerService$BackupHandler;
+ 
+     invoke-virtual {v3, v0}, Lcom/android/server/BackupManagerService$BackupHandler;->sendMessage(Landroid/os/Message;)Z
+ 
+     .end local v0           #msg:Landroid/os/Message;
+     .end local v2           #verb:I
+     :goto_1
+     monitor-exit v4
+ 
+     const/4 v3, 0x1
+ 
+     return v3
+ 
+     :cond_0
+     const/16 v2, 0xa
+ 
+     goto :goto_0
+ 
+     :cond_1
+     const-string v3, "BackupManagerService"
+ 
+     const-string v5, "Attempted to ack full backup/restore with invalid token"
+ 
+     invoke-static {v3, v5}, Landroid/util/Slog;->w(Ljava/lang/String;Ljava/lang/String;)I
+ 
+     goto :goto_1
+ 
+     .end local v1           #params:Lcom/android/server/BackupManagerService$FullParams;
+     :catchall_0
+     move-exception v3
+ 
+     monitor-exit v4
+     :try_end_0
+     .catchall {:try_start_0 .. :try_end_0} :catchall_0
+ 
+     throw v3
+ .end method
+ 
+ .method doNothing(ILcom/android/server/BackupManagerService$FullParams;)V
+     .locals 0
+     .parameter "token"
+     .parameter "params"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     return-void
+ .end method
