*** ./smali/com/android/server/LightsService$Light.smali	2013-03-18 17:22:40.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/services.jar.out/./smali/com/android/server/LightsService$Light.smali	2013-03-18 17:25:25.000000000 +0100
***************
*** 1,4 ****
! .class public final Lcom/android/server/LightsService$Light;
  .super Ljava/lang/Object;
  .source "LightsService.java"
  
--- 1,4 ----
! .class public Lcom/android/server/LightsService$Light;
  .super Ljava/lang/Object;
  .source "LightsService.java"
  
***************
*** 9,15 ****
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x11
      name = "Light"
  .end annotation
  
--- 9,15 ----
  .end annotation
  
  .annotation system Ldalvik/annotation/InnerClass;
!     accessFlags = 0x1
      name = "Light"
  .end annotation
  
***************
*** 31,37 ****
  
  
  # direct methods
! .method private constructor <init>(Lcom/android/server/LightsService;I)V
      .locals 0
      .parameter
      .parameter "id"
--- 31,37 ----
  
  
  # direct methods
! .method constructor <init>(Lcom/android/server/LightsService;I)V
      .locals 0
      .parameter
      .parameter "id"
***************
*** 68,74 ****
      return-void
  .end method
  
! .method private setLightLocked(IIIII)V
      .locals 7
      .parameter "color"
      .parameter "mode"
--- 68,74 ----
      return-void
  .end method
  
! .method setLightLocked(IIIII)V
      .locals 7
      .parameter "color"
      .parameter "mode"
***************
*** 147,153 ****
  
      move-object v0, p0
  
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 147,153 ----
  
      move-object v0, p0
  
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 207,213 ****
  
      move v3, p2
  
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      iget-object v0, p0, Lcom/android/server/LightsService$Light;->this$0:Lcom/android/server/LightsService;
  
--- 207,213 ----
  
      move v3, p2
  
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      iget-object v0, p0, Lcom/android/server/LightsService$Light;->this$0:Lcom/android/server/LightsService;
  
***************
*** 294,300 ****
      move v5, p2
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 294,300 ----
      move v5, p2
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 330,336 ****
      move v1, p1
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 330,336 ----
      move v1, p1
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 369,375 ****
      move v4, p4
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 369,375 ----
      move v4, p4
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
***************
*** 404,410 ****
      move-object v0, p0
  
      :try_start_0
!     invoke-direct/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
--- 404,410 ----
      move-object v0, p0
  
      :try_start_0
!     invoke-virtual/range {v0 .. v5}, Lcom/android/server/LightsService$Light;->setLightLocked(IIIII)V
  
      monitor-exit p0
  
