*** ./smali/com/android/server/ServerThread.smali	2013-03-18 17:22:37.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/services.jar.out/./smali/com/android/server/ServerThread.smali	2013-03-18 17:25:20.000000000 +0100
***************
*** 470,480 ****
  
      invoke-static {v3, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v63, Lcom/android/server/LightsService;
  
      move-object/from16 v0, v63
  
!     invoke-direct {v0, v4}, Lcom/android/server/LightsService;-><init>(Landroid/content/Context;)V
      :try_end_5
      .catch Ljava/lang/RuntimeException; {:try_start_5 .. :try_end_5} :catch_1
  
--- 470,480 ----
  
      invoke-static {v3, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v63, Lcom/android/server/MiuiLightsService;
  
      move-object/from16 v0, v63
  
!     invoke-direct {v0, v4}, Lcom/android/server/MiuiLightsService;-><init>(Landroid/content/Context;)V
      :try_end_5
      .catch Ljava/lang/RuntimeException; {:try_start_5 .. :try_end_5} :catch_1
  
***************
*** 922,928 ****
  
      invoke-static {v3, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v8, Lcom/android/server/net/NetworkPolicyManagerService;
  
      invoke-static {}, Lcom/android/server/am/ActivityManagerService;->self()Lcom/android/server/am/ActivityManagerService;
  
--- 922,928 ----
  
      invoke-static {v3, v9}, Landroid/util/Slog;->i(Ljava/lang/String;Ljava/lang/String;)I
  
!     new-instance v8, Lcom/android/server/net/MiuiNetworkPolicyManagerService;
  
      invoke-static {}, Lcom/android/server/am/ActivityManagerService;->self()Lcom/android/server/am/ActivityManagerService;
  
***************
*** 932,938 ****
  
      move-object v11, v6
  
!     invoke-direct/range {v8 .. v13}, Lcom/android/server/net/NetworkPolicyManagerService;-><init>(Landroid/content/Context;Landroid/app/IActivityManager;Landroid/os/IPowerManager;Landroid/net/INetworkStatsService;Landroid/os/INetworkManagementService;)V
      :try_end_19
      .catch Ljava/lang/Throwable; {:try_start_19 .. :try_end_19} :catch_d
  
--- 932,938 ----
  
      move-object v11, v6
  
!     invoke-direct/range {v8 .. v13}, Lcom/android/server/net/MiuiNetworkPolicyManagerService;-><init>(Landroid/content/Context;Landroid/app/IActivityManager;Landroid/os/IPowerManager;Landroid/net/INetworkStatsService;Landroid/os/INetworkManagementService;)V
      :try_end_19
      .catch Ljava/lang/Throwable; {:try_start_19 .. :try_end_19} :catch_d
  
