*** ./smali/android/app/ContextImpl$11.smali	2013-03-18 17:22:54.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/app/ContextImpl$11.smali	2013-03-18 17:25:42.000000000 +0100
***************
*** 31,38 ****
      .parameter "ctx"
  
      .prologue
-     new-instance v0, Landroid/app/DownloadManager;
- 
      invoke-virtual {p1}, Landroid/app/ContextImpl;->getContentResolver()Landroid/content/ContentResolver;
  
      move-result-object v1
--- 31,36 ----
***************
*** 41,47 ****
  
      move-result-object v2
  
!     invoke-direct {v0, v1, v2}, Landroid/app/DownloadManager;-><init>(Landroid/content/ContentResolver;Ljava/lang/String;)V
  
      return-object v0
  .end method
--- 39,47 ----
  
      move-result-object v2
  
!     invoke-static {v1, v2}, Landroid/util/MiuiClassFactory;->newDownloadManager(Landroid/content/ContentResolver;Ljava/lang/String;)Landroid/app/DownloadManager;
! 
!     move-result-object v0
  
      return-object v0
  .end method
