*** ./smali/android/app/ContextImpl.smali	2013-03-18 17:22:57.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/app/ContextImpl.smali	2013-03-18 17:25:46.000000000 +0100
***************
*** 1105,1110 ****
--- 1105,1112 ----
      .restart local v3       #rd:Landroid/content/IIntentReceiver;
      :cond_1
      :goto_0
+     invoke-direct {p0, p2}, Landroid/app/ContextImpl;->checkPriority(Landroid/content/IntentFilter;)V
+ 
      :try_start_0
      invoke-static {}, Landroid/app/ActivityManagerNative;->getDefault()Landroid/app/IActivityManager;
  
***************
*** 3412,3418 ****
  .end method
  
  .method final init(Landroid/app/LoadedApk;Landroid/os/IBinder;Landroid/app/ActivityThread;Landroid/content/res/Resources;Ljava/lang/String;)V
!     .locals 2
      .parameter "packageInfo"
      .parameter "activityToken"
      .parameter "mainThread"
--- 3414,3420 ----
  .end method
  
  .method final init(Landroid/app/LoadedApk;Landroid/os/IBinder;Landroid/app/ActivityThread;Landroid/content/res/Resources;Ljava/lang/String;)V
!     .locals 3
      .parameter "packageInfo"
      .parameter "activityToken"
      .parameter "mainThread"
***************
*** 3462,3476 ****
  
      iget-object v0, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
  
!     invoke-virtual {v0}, Landroid/app/LoadedApk;->getResDir()Ljava/lang/String;
  
!     move-result-object v0
  
!     invoke-virtual {p4}, Landroid/content/res/Resources;->getCompatibilityInfo()Landroid/content/res/CompatibilityInfo;
  
      move-result-object v1
  
!     invoke-virtual {p3, v0, v1}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
  
      move-result-object v0
  
--- 3464,3482 ----
  
      iget-object v0, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
  
!     iget-object v0, v0, Landroid/app/LoadedApk;->mPackageName:Ljava/lang/String;
  
!     iget-object v1, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
  
!     invoke-virtual {v1}, Landroid/app/LoadedApk;->getResDir()Ljava/lang/String;
  
      move-result-object v1
  
!     invoke-virtual {p4}, Landroid/content/res/Resources;->getCompatibilityInfo()Landroid/content/res/CompatibilityInfo;
! 
!     move-result-object v2
! 
!     invoke-virtual {p3, v0, v1, v2}, Landroid/app/ActivityThread;->getTopLevelResources(Ljava/lang/String;Ljava/lang/String;Landroid/content/res/CompatibilityInfo;)Landroid/content/res/Resources;
  
      move-result-object v0
  
***************
*** 5007,5009 ****
--- 5013,5077 ----
  
      goto :goto_0
  .end method
+ 
+ .method private checkPriority(Landroid/content/IntentFilter;)V
+     .locals 3
+     .parameter "filter"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v1, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
+ 
+     if-eqz v1, :cond_0
+ 
+     iget-object v1, p0, Landroid/app/ContextImpl;->mPackageInfo:Landroid/app/LoadedApk;
+ 
+     invoke-virtual {v1}, Landroid/app/LoadedApk;->getApplicationInfo()Landroid/content/pm/ApplicationInfo;
+ 
+     move-result-object v0
+ 
+     .local v0, ai:Landroid/content/pm/ApplicationInfo;
+     if-eqz v0, :cond_0
+ 
+     iget v1, v0, Landroid/content/pm/ApplicationInfo;->flags:I
+ 
+     and-int/lit8 v1, v1, 0x1
+ 
+     if-nez v1, :cond_0
+ 
+     invoke-virtual {p1}, Landroid/content/IntentFilter;->getPriority()I
+ 
+     move-result v1
+ 
+     const/16 v2, 0x3e8
+ 
+     if-lt v1, v2, :cond_1
+ 
+     const/16 v1, 0x3e7
+ 
+     invoke-virtual {p1, v1}, Landroid/content/IntentFilter;->setPriority(I)V
+ 
+     .end local v0           #ai:Landroid/content/pm/ApplicationInfo;
+     :cond_0
+     :goto_0
+     return-void
+ 
+     .restart local v0       #ai:Landroid/content/pm/ApplicationInfo;
+     :cond_1
+     invoke-virtual {p1}, Landroid/content/IntentFilter;->getPriority()I
+ 
+     move-result v1
+ 
+     const/16 v2, -0x3e8
+ 
+     if-gt v1, v2, :cond_0
+ 
+     const/16 v1, -0x3e7
+ 
+     invoke-virtual {p1, v1}, Landroid/content/IntentFilter;->setPriority(I)V
+ 
+     goto :goto_0
+ .end method
+ 
