*** ./smali/android/net/http/AndroidHttpClient.smali	2013-03-18 17:23:41.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/net/http/AndroidHttpClient.smali	2013-03-18 17:26:27.000000000 +0100
***************
*** 498,504 ****
  
      .local v3, sessionCache:Landroid/net/SSLSessionCache;
      :goto_0
!     invoke-static {v1, p0}, Lorg/apache/http/params/HttpProtocolParams;->setUserAgent(Lorg/apache/http/params/HttpParams;Ljava/lang/String;)V
  
      new-instance v2, Lorg/apache/http/conn/scheme/SchemeRegistry;
  
--- 498,508 ----
  
      .local v3, sessionCache:Landroid/net/SSLSessionCache;
      :goto_0
!     invoke-static {p0}, Landroid/net/http/AndroidHttpClient;->getUserAgent(Ljava/lang/String;)Ljava/lang/String;
! 
!     move-result-object v4
! 
!     invoke-static {v1, v4}, Lorg/apache/http/params/HttpProtocolParams;->setUserAgent(Lorg/apache/http/params/HttpParams;Ljava/lang/String;)V
  
      new-instance v2, Lorg/apache/http/conn/scheme/SchemeRegistry;
  
***************
*** 1197,1199 ****
--- 1201,1226 ----
  
      return-object v0
  .end method
+ 
+ .method private static getUserAgent(Ljava/lang/String;)Ljava/lang/String;
+     .locals 0
+     .parameter "userAgent"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     if-eqz p0, :cond_0
+ 
+     .end local p0
+     :goto_0
+     return-object p0
+ 
+     .restart local p0
+     :cond_0
+     invoke-static {}, Lcom/android/internal/os/RuntimeInit;->getDefaultUserAgent()Ljava/lang/String;
+ 
+     move-result-object p0
+ 
+     goto :goto_0
+ .end method
