*** ./smali/com/android/server/am/AppNotRespondingDialog$1.smali	2013-03-18 17:22:39.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/services.jar.out/./smali/com/android/server/am/AppNotRespondingDialog$1.smali	2013-03-18 17:25:23.000000000 +0100
***************
*** 38,43 ****
--- 38,60 ----
      .parameter "msg"
  
      .prologue
+     const/4 v5, 0x0
+ 
+     iget-object v3, p0, Lcom/android/server/am/AppNotRespondingDialog$1;->this$0:Lcom/android/server/am/AppNotRespondingDialog;
+ 
+     invoke-virtual {v3}, Lcom/android/server/am/AppNotRespondingDialog;->getContext()Landroid/content/Context;
+ 
+     move-result-object v3
+ 
+     iget-object v4, p0, Lcom/android/server/am/AppNotRespondingDialog$1;->this$0:Lcom/android/server/am/AppNotRespondingDialog;
+ 
+     #getter for: Lcom/android/server/am/AppNotRespondingDialog;->mProc:Lcom/android/server/am/ProcessRecord;
+     invoke-static {v4}, Lcom/android/server/am/AppNotRespondingDialog;->access$000(Lcom/android/server/am/AppNotRespondingDialog;)Lcom/android/server/am/ProcessRecord;
+ 
+     move-result-object v4
+ 
+     invoke-static {v3, v4, v5}, Lcom/android/server/am/MiuiErrorReport;->sendAnrErrorReport(Landroid/content/Context;Lcom/android/server/am/ProcessRecord;Z)V
+ 
      const/4 v1, 0x0
  
      .local v1, appErrorIntent:Landroid/content/Intent;
