*** ./smali/android/net/NetworkPolicyManager.smali	2013-03-18 17:23:34.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/net/NetworkPolicyManager.smali	2013-03-18 17:26:23.000000000 +0100
***************
*** 57,65 ****
  
      new-instance v2, Landroid/text/format/Time;
  
!     const-string v3, "UTC"
! 
!     invoke-direct {v2, v3}, Landroid/text/format/Time;-><init>(Ljava/lang/String;)V
  
      .local v2, now:Landroid/text/format/Time;
      invoke-virtual {v2, p0, p1}, Landroid/text/format/Time;->set(J)V
--- 57,63 ----
  
      new-instance v2, Landroid/text/format/Time;
  
!     invoke-direct {v2}, Landroid/text/format/Time;-><init>()V
  
      .local v2, now:Landroid/text/format/Time;
      invoke-virtual {v2, p0, p1}, Landroid/text/format/Time;->set(J)V
***************
*** 133,141 ****
  
      new-instance v2, Landroid/text/format/Time;
  
!     const-string v3, "UTC"
! 
!     invoke-direct {v2, v3}, Landroid/text/format/Time;-><init>(Ljava/lang/String;)V
  
      .local v2, now:Landroid/text/format/Time;
      invoke-virtual {v2, p0, p1}, Landroid/text/format/Time;->set(J)V
--- 131,137 ----
  
      new-instance v2, Landroid/text/format/Time;
  
!     invoke-direct {v2}, Landroid/text/format/Time;-><init>()V
  
      .local v2, now:Landroid/text/format/Time;
      invoke-virtual {v2, p0, p1}, Landroid/text/format/Time;->set(J)V
