*** ./smali/android/content/pm/PackageManager.smali	2013-03-18 17:23:22.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/content/pm/PackageManager.smali	2013-03-18 17:26:16.000000000 +0100
***************
*** 14,25 ****
--- 14,29 ----
  # static fields
  .field public static final ACTION_CLEAN_EXTERNAL_STORAGE:Ljava/lang/String; = "android.content.pm.CLEAN_EXTERNAL_STORAGE"
  
+ .field public static final COMPONENT_ENABLED_STATE_ACCESS_CONTROL:I = -0x80000000
+ 
  .field public static final COMPONENT_ENABLED_STATE_DEFAULT:I = 0x0
  
  .field public static final COMPONENT_ENABLED_STATE_DISABLED:I = 0x2
  
  .field public static final COMPONENT_ENABLED_STATE_DISABLED_USER:I = 0x3
  
+ .field public static final COMPONENT_ENABLED_STATE_DISABLE_AUTOSTART:I = 0x40000000
+ 
  .field public static final COMPONENT_ENABLED_STATE_ENABLED:I = 0x1
  
  .field public static final DELETE_FAILED_DEVICE_POLICY_MANAGER:I = -0x2
***************
*** 272,277 ****
--- 276,283 ----
  
  .field public static final VERIFICATION_REJECT:I = -0x1
  
+ .field public static final HAS_ACTIVITY:I = 0x20000
+ 
  
  # direct methods
  .method public constructor <init>()V
