*** ./smali/android/content/pm/ResolveInfo.smali	2013-03-18 17:23:22.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/content/pm/ResolveInfo.smali	2013-03-18 17:26:15.000000000 +0100
***************
*** 646,656 ****
  
      iget-object v3, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
  
-     iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
- 
      const/4 v5, 0x0
  
!     invoke-virtual {p1, v3, v4, v5}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
--- 645,653 ----
  
      iget-object v3, p0, Landroid/content/pm/ResolveInfo;->resolvePackageName:Ljava/lang/String;
  
      const/4 v5, 0x0
  
!     invoke-direct {p0, p1, v3, v5}, Landroid/content/pm/ResolveInfo;->loadDrawableFromTheme(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
***************
*** 680,688 ****
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
!     iget v4, p0, Landroid/content/pm/ResolveInfo;->icon:I
! 
!     invoke-virtual {p1, v3, v4, v0}, Landroid/content/pm/PackageManager;->getDrawable(Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
--- 677,683 ----
  
      iget-object v3, v1, Landroid/content/pm/ComponentInfo;->packageName:Ljava/lang/String;
  
!     invoke-direct {p0, p1, v3, v0}, Landroid/content/pm/ResolveInfo;->loadDrawableFromTheme(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
  
      move-result-object v2
  
***************
*** 1027,1029 ****
--- 1022,1069 ----
  
      goto :goto_2
  .end method
+ 
+ .method private loadDrawableFromTheme(Landroid/content/pm/PackageManager;Ljava/lang/String;Landroid/content/pm/ApplicationInfo;)Landroid/graphics/drawable/Drawable;
+     .locals 6
+     .parameter "pm"
+     .parameter "packageName"
+     .parameter "ai"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v0, p0, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     if-eqz v0, :cond_0
+ 
+     iget-object v4, p0, Landroid/content/pm/ResolveInfo;->activityInfo:Landroid/content/pm/ActivityInfo;
+ 
+     .local v4, ci:Landroid/content/pm/ComponentInfo;
+     :goto_0
+     iget v2, p0, Landroid/content/pm/ResolveInfo;->icon:I
+ 
+     iget-object v0, p0, Landroid/content/pm/ResolveInfo;->filter:Landroid/content/IntentFilter;
+ 
+     invoke-static {v0}, Landroid/app/MiuiThemeHelper;->isCustomizedIcon(Landroid/content/IntentFilter;)Z
+ 
+     move-result v5
+ 
+     move-object v0, p1
+ 
+     move-object v1, p2
+ 
+     move-object v3, p3
+ 
+     invoke-static/range {v0 .. v5}, Landroid/app/MiuiThemeHelper;->getDrawable(Landroid/content/pm/PackageManager;Ljava/lang/String;ILandroid/content/pm/ApplicationInfo;Landroid/content/pm/PackageItemInfo;Z)Landroid/graphics/drawable/Drawable;
+ 
+     move-result-object v0
+ 
+     return-object v0
+ 
+     .end local v4           #ci:Landroid/content/pm/ComponentInfo;
+     :cond_0
+     iget-object v4, p0, Landroid/content/pm/ResolveInfo;->serviceInfo:Landroid/content/pm/ServiceInfo;
+ 
+     goto :goto_0
+ .end method
\ No newline at end of file
