*** ./smali/android/media/MediaScanner$MyMediaScannerClient.smali	2013-03-18 17:23:30.000000000 +0100
--- /root/miui_sony_xperia/stock_lotus/temp/new_smali/framework.jar.out/./smali/android/media/MediaScanner$MyMediaScannerClient.smali	2013-03-18 17:26:20.000000000 +0100
***************
*** 661,675 ****
      :cond_e
      const-string v24, "notification_sound"
  
      move-object/from16 v0, p0
  
      move-object/from16 v1, v24
  
!     move-object/from16 v2, v19
  
!     move-wide/from16 v3, v17
  
!     invoke-direct {v0, v1, v2, v3, v4}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Landroid/net/Uri;J)V
  
      move-object/from16 v0, p0
  
--- 661,711 ----
      :cond_e
      const-string v24, "notification_sound"
  
+     move-object/from16 v0, p1
+ 
+     iget-object v0, v0, Landroid/media/MediaScanner$FileCacheEntry;->mPath:Ljava/lang/String;
+ 
+     move-object/from16 v25, v0
+ 
+     move-object/from16 v0, p0
+ 
+     move-object/from16 v1, v24
+ 
+     move-object/from16 v2, v25
+ 
+     invoke-direct {v0, v1, v2}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Ljava/lang/String;)V
+ 
+     const-string v24, "sms_delivered_sound"
+ 
+     move-object/from16 v0, p1
+ 
+     iget-object v0, v0, Landroid/media/MediaScanner$FileCacheEntry;->mPath:Ljava/lang/String;
+ 
+     move-object/from16 v25, v0
+ 
      move-object/from16 v0, p0
  
      move-object/from16 v1, v24
  
!     move-object/from16 v2, v25
! 
!     invoke-direct {v0, v1, v2}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Ljava/lang/String;)V
! 
!     const-string v24, "sms_received_sound"
! 
!     move-object/from16 v0, p1
! 
!     iget-object v0, v0, Landroid/media/MediaScanner$FileCacheEntry;->mPath:Ljava/lang/String;
! 
!     move-object/from16 v25, v0
! 
!     move-object/from16 v0, p0
! 
!     move-object/from16 v1, v24
  
!     move-object/from16 v2, v25
  
!     invoke-direct {v0, v1, v2}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Ljava/lang/String;)V
  
      move-object/from16 v0, p0
  
***************
*** 1096,1110 ****
      :cond_1a
      const-string v24, "ringtone"
  
      move-object/from16 v0, p0
  
      move-object/from16 v1, v24
  
!     move-object/from16 v2, v19
! 
!     move-wide/from16 v3, v17
  
!     invoke-direct {v0, v1, v2, v3, v4}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Landroid/net/Uri;J)V
  
      move-object/from16 v0, p0
  
--- 1132,1150 ----
      :cond_1a
      const-string v24, "ringtone"
  
+     move-object/from16 v0, p1
+ 
+     iget-object v0, v0, Landroid/media/MediaScanner$FileCacheEntry;->mPath:Ljava/lang/String;
+ 
+     move-object/from16 v25, v0
+ 
      move-object/from16 v0, p0
  
      move-object/from16 v1, v24
  
!     move-object/from16 v2, v25
  
!     invoke-direct {v0, v1, v2}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Ljava/lang/String;)V
  
      move-object/from16 v0, p0
  
***************
*** 1197,1211 ****
      :cond_1c
      const-string v24, "alarm_alert"
  
      move-object/from16 v0, p0
  
      move-object/from16 v1, v24
  
!     move-object/from16 v2, v19
! 
!     move-wide/from16 v3, v17
  
!     invoke-direct {v0, v1, v2, v3, v4}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Landroid/net/Uri;J)V
  
      move-object/from16 v0, p0
  
--- 1237,1255 ----
      :cond_1c
      const-string v24, "alarm_alert"
  
+     move-object/from16 v0, p1
+ 
+     iget-object v0, v0, Landroid/media/MediaScanner$FileCacheEntry;->mPath:Ljava/lang/String;
+ 
+     move-object/from16 v25, v0
+ 
      move-object/from16 v0, p0
  
      move-object/from16 v1, v24
  
!     move-object/from16 v2, v25
  
!     invoke-direct {v0, v1, v2}, Landroid/media/MediaScanner$MyMediaScannerClient;->setSettingIfNotSet(Ljava/lang/String;Ljava/lang/String;)V
  
      move-object/from16 v0, p0
  
***************
*** 3011,3013 ****
--- 3055,3112 ----
  
      goto :goto_0
  .end method
+ 
+ 
+ .method private setSettingIfNotSet(Ljava/lang/String;Ljava/lang/String;)V
+     .locals 3
+     .parameter "settingName"
+     .parameter "path"
+     .annotation build Landroid/annotation/MiuiHook;
+         value = .enum Landroid/annotation/MiuiHook$MiuiHookType;->NEW_METHOD:Landroid/annotation/MiuiHook$MiuiHookType;
+     .end annotation
+ 
+     .prologue
+     iget-object v1, p0, Landroid/media/MediaScanner$MyMediaScannerClient;->this$0:Landroid/media/MediaScanner;
+ 
+     iget-object v1, v1, Landroid/media/MediaScanner;->mContext:Landroid/content/Context;
+ 
+     invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
+ 
+     move-result-object v1
+ 
+     invoke-static {v1, p1}, Landroid/provider/Settings$System;->getString(Landroid/content/ContentResolver;Ljava/lang/String;)Ljava/lang/String;
+ 
+     move-result-object v0
+ 
+     .local v0, existingSettingValue:Ljava/lang/String;
+     invoke-static {v0}, Landroid/text/TextUtils;->isEmpty(Ljava/lang/CharSequence;)Z
+ 
+     move-result v1
+ 
+     if-eqz v1, :cond_0
+ 
+     iget-object v1, p0, Landroid/media/MediaScanner$MyMediaScannerClient;->this$0:Landroid/media/MediaScanner;
+ 
+     iget-object v1, v1, Landroid/media/MediaScanner;->mContext:Landroid/content/Context;
+ 
+     invoke-virtual {v1}, Landroid/content/Context;->getContentResolver()Landroid/content/ContentResolver;
+ 
+     move-result-object v1
+ 
+     new-instance v2, Ljava/io/File;
+ 
+     invoke-direct {v2, p2}, Ljava/io/File;-><init>(Ljava/lang/String;)V
+ 
+     invoke-static {v2}, Landroid/net/Uri;->fromFile(Ljava/io/File;)Landroid/net/Uri;
+ 
+     move-result-object v2
+ 
+     invoke-virtual {v2}, Landroid/net/Uri;->toString()Ljava/lang/String;
+ 
+     move-result-object v2
+ 
+     invoke-static {v1, p1, v2}, Landroid/provider/Settings$System;->putString(Landroid/content/ContentResolver;Ljava/lang/String;Ljava/lang/String;)Z
+ 
+     :cond_0
+     return-void
+ .end method
